src/ConnectedMachine/readme.md

```diff
@@ -40,45 +40,108 @@ In this directory, run AutoRest:
 > see https://aka.ms/autorest
 
 ``` yaml
-powershell: true
-azure: true
-branch: master
-repo: https://github.com/Azure/azure-rest-api-specs/tree/$(branch)
-metadata:
-  authors: Microsoft Corporation
-  owners: Microsoft Corporation
-  description: 'Microsoft Azure PowerShell: $(service-name) cmdlets'
-  copyright: Microsoft Corporation. All rights reserved.
-  tags: Azure ResourceManager ARM PSModule $(service-name)
-  companyName: Microsoft Corporation
-  requireLicenseAcceptance: true
-  licenseUri: https://aka.ms/azps-license
-  projectUri: https://github.com/Azure/azure-powershell
-prefix: Az
-subject-prefix: 'Connected'
-module-name: $(prefix).$(service-name)
-namespace: Microsoft.Azure.PowerShell.Cmdlets.$(service-name)
-clear-output-folder: true
-output-folder: .
-input-file:
-  - $(repo)/specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2019-08-02/HybridCompute.json
-
+require:
+  - $(this-folder)/../readme.azure.noprofile.md
 module-version: 0.1.0
 title: ConnectedMachine
+subject-prefix: 'Connected'
+input-file:
+  - $(repo)/specification/hybridcompute/resource-manager/Microsoft.HybridCompute/stable/2019-12-12/HybridCompute.json
 
 directive:
   - where:
       subject: Operation
     hide: true
   - where: $.definitions.Identifier.properties
     suppress: R3019
-  - remove-operation: 
+
+  # GetViaIdentity isn't useful until Azure PowerShell supports piping of different subjects
+  - where:
+      verb: New
+      variant: ^GetViaIdentity\d?$
+    remove: true
+
+  # Set correct variants for PUT and PATCH verbs
+  - where:
+      verb: New
+      variant: ^CreateViaIdentity\d?$|^Create\d?$
+    remove: true
+  - where:
```

Comment:
Update/Set cmdlets only support "by Name" parameter here?

Reply:
Yeah without those lines, you get this parameter set (both for `Set` and `Update`) but it's not very useful:

> Set-AzConnectedMachineExtension -MachineName <string> -Name <string> -ResourceGroupName <string> -ExtensionParameter <IMachineExtension> [-SubscriptionId <string>] [-DefaultProfile <psobject>] [-AsJob] [-Break] [-HttpPipelineAppend <SendAsyncStep[]>] [-HttpPipelinePrepend <SendAsyncStep[]>] [-NoWait] [-Proxy <uri>] [-ProxyCredential <pscredential>] [-ProxyUseDefaultCredentials] [-WhatIf] [-Confirm] [<CommonParameters>]

I guess it would allow you to get an extension on one machine and maybe set it on another machine... that might have some use... I'll see if it works.

Ideally I'd want to light up the scenario where you can pipe `Get-AzConnectedMachine | Set-AzConnectedMachineExtension` so that I can get my whole fleet of machines and then add an extension to all of them, but that doesn't seem to be possible. I've tried running that and it gives me:

```
Exception: The running command stopped because the preference variable "ErrorActionPreference" or common parameter is set to Stop: Invalid identity for URI '/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.HybridCompute/machines/{name}/extensions/{extensionName}'
```

Reply:
Can run the following cmdlet with `-debug` parameter?

```
$ext | Set-AzConnectedMachineExtension -MachineName (hostname) -ResourceGroupName arc-server-test -Name foo -debug
```

Reply:
```
$a | Set-AzConnectedMachineExtension -MachineName (hostname) -ResourceGroupName arc-server-test -Name foo -Debug
DEBUG: 10:08:14 PM - GetAzureRMContextCommand begin processing with ParameterSet 'GetSingleContext'.
DEBUG: 10:08:14 PM - using account id 'tyleonha@microsoft.com'...
DEBUG: AzureQoSEvent: CommandName - Get-AzContext; IsSuccess - True; Duration - 00:00:00.0020808;
DEBUG: Finish sending metric.
DEBUG: 10:08:15 PM - GetAzureRMContextCommand end processing.
VERBOSE: INFORMATION: CMDLET BEGIN PROCESSING The contents are 'CmdletBeginProcessing' and ''

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): y
VERBOSE: INFORMATION: CMDLET BEGIN PROCESSING Parameter: ''
CMDLET BEGIN PROCESSING RequestMessage ''
CMDLET BEGIN PROCESSING Response: ''
CMDLET BEGIN PROCESSING Value: '0'

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
VERBOSE: INFORMATION: CMDLET BEGIN PROCESSING ExtendedData Type: ''
CMDLET BEGIN PROCESSING ExtendedData ''

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
DEBUG: CmdletBeginProcessing: 
VERBOSE: INFORMATION: CMDLET PROCESS RECORD START The contents are 'CmdletProcessRecordStart' and ''

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
VERBOSE: INFORMATION: CMDLET PROCESS RECORD START Parameter: ''
CMDLET PROCESS RECORD START RequestMessage ''
CMDLET PROCESS RECORD START Response: ''
CMDLET PROCESS RECORD START Value: '0'

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
VERBOSE: INFORMATION: CMDLET PROCESS RECORD START ExtendedData Type: ''
CMDLET PROCESS RECORD START ExtendedData ''

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
DEBUG: CmdletProcessRecordStart: 

Confirm
Are you sure you want to perform this action?
Performing the operation "Set-AzConnectedMachineExtension_Update" on target "Call remote 'MachineExtensionsCreateOrUpdate' operation".
[Y] Yes  [A] Yes to All  [N] No  [L] No to All  [S] Suspend  [?] Help (default is "Y"): 
DEBUG: [CmdletProcessRecordAsyncStart]: Created new QosEvent for command 'Set-AzConnectedMachineExtension_Update': AzureQoSEvent: CommandName - Set-AzConnectedMachineExtension_Update; IsSuccess - False; Duration - 00:00:00;
DEBUG: CmdletProcessRecordAsyncStart: 
VERBOSE: INFORMATION: CMDLET GET PIPELINE The contents are 'CmdletGetPipeline' and ''

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
VERBOSE: INFORMATION: CMDLET GET PIPELINE Parameter: ''
CMDLET GET PIPELINE RequestMessage ''
CMDLET GET PIPELINE Response: ''
CMDLET GET PIPELINE Value: '0'

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
VERBOSE: INFORMATION: CMDLET GET PIPELINE ExtendedData Type: ''
CMDLET GET PIPELINE ExtendedData ''

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
DEBUG: CmdletGetPipeline: 
VERBOSE: INFORMATION: CMDLET BEFORE API CALL The contents are 'CmdletBeforeAPICall' and ''

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
VERBOSE: INFORMATION: CMDLET BEFORE API CALL Parameter: ''
CMDLET BEFORE API CALL RequestMessage ''
CMDLET BEFORE API CALL Response: ''
CMDLET BEFORE API CALL Value: '0'

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
VERBOSE: INFORMATION: CMDLET BEFORE API CALL ExtendedData Type: ''
CMDLET BEFORE API CALL ExtendedData ''

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
DEBUG: CmdletBeforeAPICall: 
VERBOSE: INFORMATION: URL CREATED The contents are 'URLCreated' and 'https://management.azure.com/subscriptions/422b6c61-95b0-4213-b3be-7282315df71d/resourceGroups/arc-server-test/providers/Microsoft.HybridCompute/machines/tyler-ThinkPad-X1-Carbon/extensions/foo?api-version=2019-12-12'

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
VERBOSE: INFORMATION: URL CREATED Parameter: ''
URL CREATED RequestMessage ''
URL CREATED Response: ''
URL CREATED Value: '0'

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
VERBOSE: INFORMATION: URL CREATED ExtendedData Type: ''
URL CREATED ExtendedData ''

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
DEBUG: URLCreated: https://management.azure.com/subscriptions/422b6c61-95b0-4213-b3be-7282315df71d/resourceGroups/arc-server-test/providers/Microsoft.HybridCompute/machines/tyler-ThinkPad-X1-Carbon/extensions/foo?api-version=2019-12-12
VERBOSE: INFORMATION: REQUEST CREATED The contents are 'RequestCreated' and 'https://management.azure.com/subscriptions/422b6c61-95b0-4213-b3be-7282315df71d/resourceGroups/arc-server-test/providers/Microsoft.HybridCompute/machines/tyler-ThinkPad-X1-Carbon/extensions/foo?api-version=2019-12-12'

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
VERBOSE: INFORMATION: REQUEST CREATED Parameter: ''
REQUEST CREATED RequestMessage ''
REQUEST CREATED Response: ''
REQUEST CREATED Value: '0'

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
VERBOSE: INFORMATION: REQUEST CREATED ExtendedData Type: ''
REQUEST CREATED ExtendedData ''

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
DEBUG: RequestCreated: https://management.azure.com/subscriptions/422b6c61-95b0-4213-b3be-7282315df71d/resourceGroups/arc-server-test/providers/Microsoft.HybridCompute/machines/tyler-ThinkPad-X1-Carbon/extensions/foo?api-version=2019-12-12
VERBOSE: INFORMATION: HEADER PARAMETERS ADDED The contents are 'HeaderParametersAdded' and 'https://management.azure.com/subscriptions/422b6c61-95b0-4213-b3be-7282315df71d/resourceGroups/arc-server-test/providers/Microsoft.HybridCompute/machines/tyler-ThinkPad-X1-Carbon/extensions/foo?api-version=2019-12-12'

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
VERBOSE: INFORMATION: HEADER PARAMETERS ADDED Parameter: ''
HEADER PARAMETERS ADDED RequestMessage ''
HEADER PARAMETERS ADDED Response: ''
HEADER PARAMETERS ADDED Value: '0'

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
VERBOSE: INFORMATION: HEADER PARAMETERS ADDED ExtendedData Type: ''
HEADER PARAMETERS ADDED ExtendedData ''

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
DEBUG: HeaderParametersAdded: https://management.azure.com/subscriptions/422b6c61-95b0-4213-b3be-7282315df71d/resourceGroups/arc-server-test/providers/Microsoft.HybridCompute/machines/tyler-ThinkPad-X1-Carbon/extensions/foo?api-version=2019-12-12
VERBOSE: INFORMATION: BODY CONTENT SET The contents are 'BodyContentSet' and 'https://management.azure.com/subscriptions/422b6c61-95b0-4213-b3be-7282315df71d/resourceGroups/arc-server-test/providers/Microsoft.HybridCompute/machines/tyler-ThinkPad-X1-Carbon/extensions/foo?api-version=2019-12-12'

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
VERBOSE: INFORMATION: BODY CONTENT SET Parameter: ''
BODY CONTENT SET RequestMessage ''
BODY CONTENT SET Response: ''
BODY CONTENT SET Value: '0'

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
VERBOSE: INFORMATION: BODY CONTENT SET ExtendedData Type: ''
BODY CONTENT SET ExtendedData ''

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
DEBUG: BodyContentSet: https://management.azure.com/subscriptions/422b6c61-95b0-4213-b3be-7282315df71d/resourceGroups/arc-server-test/providers/Microsoft.HybridCompute/machines/tyler-ThinkPad-X1-Carbon/extensions/foo?api-version=2019-12-12
DEBUG: ============================ HTTP REQUEST ============================

HTTP Method:
PUT

Absolute Uri:
https://management.azure.com/subscriptions/422b6c61-95b0-4213-b3be-7282315df71d/resourceGroups/arc-server-test/providers/Microsoft.HybridCompute/machines/tyler-ThinkPad-X1-Carbon/extensions/foo?api-version=2019-12-12

Headers:

Body:
{
  "location": "eastus",
  "properties": {
    "instanceView": {
      "status": {
        "code": "success",
        "level": "Information",
        "message": "formattedMessage: Enable succeeded: \n[stdout]\nstderr\nstdout\n\n[stderr]\n"
      },
      "name": "foo",
      "type": "CustomScript",
      "typeHandlerVersion": "2.1.3"
    },
    "type": "CustomScript",
    "autoUpgradeMinorVersion": false,
    "publisher": "Microsoft.Azure.Extensions",
    "settings": {
      "commandToExecute": "ls -a"
    },
    "typeHandlerVersion": "2.1.3"
  }
}


DEBUG: BeforeCall: 
DEBUG: [ADAL]: Information: 2020-08-26T05:10:02.1032164Z: 073edd79-58ef-43e5-ab14-8a8cecda349e - LoggerBase.cs: ADAL PCL.CoreCLR with assembly version '3.19.2.6005', file version '3.19.50302.0130' and informational version '2a8bec6c4c76d0c1ef819b55bdc3cda2d2605056' is running...
DEBUG: [ADAL]: Information: 2020-08-26T05:10:02.1033759Z: 073edd79-58ef-43e5-ab14-8a8cecda349e - LoggerBase.cs: ADAL PCL.CoreCLR with assembly version '3.19.2.6005', file version '3.19.50302.0130' and informational version '2a8bec6c4c76d0c1ef819b55bdc3cda2d2605056' is running...
DEBUG: [ADAL]: Information: 2020-08-26T05:10:02.1034778Z: 073edd79-58ef-43e5-ab14-8a8cecda349e - LoggerBase.cs: === Token Acquisition started: 
	CacheType: null
	Authentication Target: User
	, Authority Host: login.microsoftonline.com
DEBUG: [ADAL]: Information: 2020-08-26T05:10:02.1035336Z: 073edd79-58ef-43e5-ab14-8a8cecda349e - LoggerBase.cs: === Token Acquisition started:
	Authority: https://login.microsoftonline.com/72f988bf-86f1-41af-91ab-2d7cd011db47/
	Resource: https://management.core.windows.net/
	ClientId: 1950a258-227b-4e31-a9cf-717495945fc2
	CacheType: null
	Authentication Target: User
	
DEBUG: [ADAL]: Verbose: 2020-08-26T05:10:02.1036685Z: 073edd79-58ef-43e5-ab14-8a8cecda349e - LoggerBase.cs: Loading from cache.
DEBUG: [ADAL]: Verbose: 2020-08-26T05:10:02.1037197Z: 073edd79-58ef-43e5-ab14-8a8cecda349e - LoggerBase.cs: Loading from cache.
DEBUG: [ADAL]: Information: 2020-08-26T05:10:02.1048311Z: 00000000-0000-0000-0000-000000000000 - LoggerBase.cs: Deserialized 3 items to token cache.
DEBUG: [ADAL]: Verbose: 2020-08-26T05:10:02.1049878Z: 073edd79-58ef-43e5-ab14-8a8cecda349e - LoggerBase.cs: Looking up cache for a token...
DEBUG: [ADAL]: Verbose: 2020-08-26T05:10:02.1050598Z: 073edd79-58ef-43e5-ab14-8a8cecda349e - LoggerBase.cs: Looking up cache for a token...
DEBUG: [ADAL]: Information: 2020-08-26T05:10:02.1051668Z: 073edd79-58ef-43e5-ab14-8a8cecda349e - LoggerBase.cs: An item matching the requested resource was found in the cache
DEBUG: [ADAL]: Information: 2020-08-26T05:10:02.1052312Z: 073edd79-58ef-43e5-ab14-8a8cecda349e - LoggerBase.cs: An item matching the requested resource was found in the cache
DEBUG: [ADAL]: Information: 2020-08-26T05:10:02.1053198Z: 073edd79-58ef-43e5-ab14-8a8cecda349e - LoggerBase.cs: 57.401494916666664 minutes left until token in cache expires
DEBUG: [ADAL]: Information: 2020-08-26T05:10:02.1053582Z: 073edd79-58ef-43e5-ab14-8a8cecda349e - LoggerBase.cs: 57.401494916666664 minutes left until token in cache expires
DEBUG: [ADAL]: Information: 2020-08-26T05:10:02.1053898Z: 073edd79-58ef-43e5-ab14-8a8cecda349e - LoggerBase.cs: A matching item (access token or refresh token or both) was found in the cache
DEBUG: [ADAL]: Information: 2020-08-26T05:10:02.1054269Z: 073edd79-58ef-43e5-ab14-8a8cecda349e - LoggerBase.cs: A matching item (access token or refresh token or both) was found in the cache
DEBUG: [ADAL]: Information: 2020-08-26T05:10:02.1055613Z: 073edd79-58ef-43e5-ab14-8a8cecda349e - LoggerBase.cs: === Token Acquisition finished successfully. An access token was returned: Expiration Time: 8/26/2020 6:07:26 AM +00:00
DEBUG: [ADAL]: Information: 2020-08-26T05:10:02.1056130Z: 073edd79-58ef-43e5-ab14-8a8cecda349e - LoggerBase.cs: === Token Acquisition finished successfully. An access token was returned: Expiration Time: 8/26/2020 6:07:26 AM +00:00Access Token Hash: XNvbFj1C+pyfghSu12Vz3fD8JfvYsZTHp4Ff75DRkDA=
	 User id: c256c7b0-e94d-49c4-8585-0c7f3020bd37
DEBUG: ============================ HTTP RESPONSE ============================

Status Code:
Accepted

Headers:
Cache-Control                 : no-cache
Pragma                        : no-cache
Location                      : https://management.azure.com/subscriptions/422b6c61-95b0-4213-b3be-7282315df71d/providers/Microsoft.HybridCompute/locations/eastus/operationresults/372f7387-0f1a-4eb0-a62a-369c779c3cc7?api-version=2019-12-12
Strict-Transport-Security     : max-age=31536000; includeSubDomains
x-ms-request-id               : 7fc67a75-d2b1-4f2d-af48-364100a83ebc
Azure-AsyncOperation          : https://management.azure.com/subscriptions/422b6c61-95b0-4213-b3be-7282315df71d/providers/Microsoft.HybridCompute/locations/eastus/operationstatus/372f7387-0f1a-4eb0-a62a-369c779c3cc7?api-version=2019-12-12
Server                        : Kestrel
x-ms-ratelimit-remaining-subscription-resource-requests: 319
x-ms-correlation-request-id   : 883efb04-8272-4591-93a5-fba69fae8faf
x-ms-routing-request-id       : WESTUS2:20200826T051005Z:883efb04-8272-4591-93a5-fba69fae8faf
X-Content-Type-Options        : nosniff
Date                          : Wed, 26 Aug 2020 05:10:04 GMT

Body:



DEBUG: ResponseCreated: 
VERBOSE: INFORMATION: DELAY BEFORE POLLING The contents are 'DelayBeforePolling' and 'Delaying 30 seconds before polling.'

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
VERBOSE: INFORMATION: DELAY BEFORE POLLING Parameter: ''
DELAY BEFORE POLLING RequestMessage 'Method: PUT, RequestUri: 'https://management.azure.com/subscriptions/422b6c61-95b0-4213-b3be-7282315df71d/resourceGroups/arc-server-test/providers/Microsoft.HybridCompute/machines/tyler-ThinkPad-X1-Carbon/extensions/foo?api-version=2019-12-12', Version: 1.1, Content: System.Net.Http.StringContent, Headers:
{
  x-ms-unique-id: 27
  x-ms-client-request-id: c7c5b66b-b95f-4d23-900b-24a34d04b320
  CommandName: Set-AzConnectedMachineExtension
  FullCommandName: Set-AzConnectedMachineExtension_Update
  ParameterSetName: __AllParameterSets
  User-Agent: AzurePowershell/Az4.0.0-preview
  Authorization: Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsIng1dCI6ImppYk5ia0ZTU2JteFBZck45Q0ZxUms0SzRndyIsImtpZCI6ImppYk5ia0ZTU2JteFBZck45Q0ZxUms0SzRndyJ9.eyJhdWQiOiJodHRwczovL21hbmFnZW1lbnQuY29yZS53aW5kb3dzLm5ldC8iLCJpc3MiOiJodHRwczovL3N0cy53aW5kb3dzLm5ldC83MmY5ODhiZi04NmYxLTQxYWYtOTFhYi0yZDdjZDAxMWRiNDcvIiwiaWF0IjoxNTk4NDE4MTQ4LCJuYmYiOjE1OTg0MTgxNDgsImV4cCI6MTU5ODQyMjA0OCwiX2NsYWltX25hbWVzIjp7Imdyb3VwcyI6InNyYzEifSwiX2NsYWltX3NvdXJjZXMiOnsic3JjMSI6eyJlbmRwb2ludCI6Imh0dHBzOi8vZ3JhcGgud2luZG93cy5uZXQvNzJmOTg4YmYtODZmMS00MWFmLTkxYWItMmQ3Y2QwMTFkYjQ3L3VzZXJzL2MyNTZjN2IwLWU5NGQtNDljNC04NTg1LTBjN2YzMDIwYmQzNy9nZXRNZW1iZXJPYmplY3RzIn19LCJhY3IiOiIxIiwiYWlvIjoiQVZRQXEvOFFBQUFBcnNiRnYvMU5kdXVpcjJuNy9nYUkzYkw4SkVNY3E1YnVlOEtZMGE3bk5VK3YwOFpIT1YzTG5EZGxqWkxrZUZRbk52OEJ5NUlMZkpIY3ErN29vaEFLQmx1bUV4Z1NNdis2OXRPVTRWa3pOcFE9IiwiYW1yIjpbInB3ZCIsIm1mYSJdLCJhcHBpZCI6IjE5NTBhMjU4LTIyN2ItNGUzMS1hOWNmLTcxNzQ5NTk0NWZjMiIsImFwcGlkYWNyIjoiMCIsImZhbWlseV9uYW1lIjoiTGVvbmhhcmR0IChQT1dFUlNIRUxMKSIsImdpdmVuX25hbWUiOiJUeWxlciIsImlwYWRkciI6Ijc1LjE3Mi4xODMuMTc2IiwibmFtZSI6IlR5bGVyIExlb25oYXJkdCAoUE9XRVJTSEVMTCkiLCJvaWQiOiJjMjU2YzdiMC1lOTRkLTQ5YzQtODU4NS0wYzdmMzAyMGJkMzciLCJvbnByZW1fc2lkIjoiUy0xLTUtMjEtMjEyNzUyMTE4NC0xNjA0MDEyOTIwLTE4ODc5Mjc1MjctMjIyNjkzMzgiLCJwdWlkIjoiMTAwMzdGRkU5OThDRkZGNyIsInJoIjoiMC5BUm9BdjRqNWN2R0dyMEdScXkxODBCSGJSMWlpVUJsN0lqRk9xYzl4ZEpXVVg4SWFBTk0uIiwic2NwIjoidXNlcl9pbXBlcnNvbmF0aW9uIiwic3ViIjoiMXJrT05YeDAtUDVJQnlFc2kyV1RROVZXQUtmNW9IQjRVRnNkV040aFlBSSIsInRpZCI6IjcyZjk4OGJmLTg2ZjEtNDFhZi05MWFiLTJkN2NkMDExZGI0NyIsInVuaXF1ZV9uYW1lIjoidHlsZW9uaGFAbWljcm9zb2Z0LmNvbSIsInVwbiI6InR5bGVvbmhhQG1pY3Jvc29mdC5jb20iLCJ1dGkiOiJZYzF3TWtsUUYwMkRiU1k5RXR3TEFBIiwidmVyIjoiMS4wIiwieG1zX3RjZHQiOjEyODkyNDE1NDd9.VbmeQcgINWvA1hIQ2IOZh1ND1QpY0qwo5UQm21Y2zXTJ6ah1PEJN4iSWOxk8KWsc7lpNB3BBEWtgoT4YDhE79kmFZl-miiw2CRTstzrYLSUVD4vG_yAhjJ6b7q3bgv_fnvioZ6TekJ7J9DNQWTLC_IVaD57-vnN8JDbG-GAD356Zl_ywBKGhZoJkoJPK-O_zb73SpP6kUw8uxa49UGqI3qt7VVAVYRIkSnPucLZu5VjwEBSGxzVpuUANh7Ag7Pz8k1ZXEupjTcsF2kcba3vVumtHm90PmpXoJlkJDQV8os7MpDaRJOcbc1p5FlzpiReL70HXxumZB2H5HmbCbucX1A
  Content-Type: application/json
  Content-Length: 555
}'
DELAY BEFORE POLLING Response: 'StatusCode: 202, ReasonPhrase: 'Accepted', Version: 1.1, Content: System.Net.Http.HttpConnectionResponseContent, Headers:
{
  Cache-Control: no-cache
  Pragma: no-cache
  Location: https://management.azure.com/subscriptions/422b6c61-95b0-4213-b3be-7282315df71d/providers/Microsoft.HybridCompute/locations/eastus/operationresults/372f7387-0f1a-4eb0-a62a-369c779c3cc7?api-version=2019-12-12
  Strict-Transport-Security: max-age=31536000; includeSubDomains
  x-ms-request-id: 7fc67a75-d2b1-4f2d-af48-364100a83ebc
  Azure-AsyncOperation: https://management.azure.com/subscriptions/422b6c61-95b0-4213-b3be-7282315df71d/providers/Microsoft.HybridCompute/locations/eastus/operationstatus/372f7387-0f1a-4eb0-a62a-369c779c3cc7?api-version=2019-12-12
  Server: Kestrel
  x-ms-ratelimit-remaining-subscription-resource-requests: 319
  x-ms-correlation-request-id: 883efb04-8272-4591-93a5-fba69fae8faf
  x-ms-routing-request-id: WESTUS2:20200826T051005Z:883efb04-8272-4591-93a5-fba69fae8faf
  X-Content-Type-Options: nosniff
  Date: Wed, 26 Aug 2020 05:10:04 GMT
  Content-Type: application/json
  Expires: -1
  Content-Length: 0
}'
DELAY BEFORE POLLING Value: '0'

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
VERBOSE: INFORMATION: DELAY BEFORE POLLING ExtendedData Type: ''
DELAY BEFORE POLLING ExtendedData ''

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
DEBUG: DelayBeforePolling: Delaying 30 seconds before polling.
VERBOSE: INFORMATION: POLLING The contents are 'Polling' and 'Polling https://management.azure.com/subscriptions/422b6c61-95b0-4213-b3be-7282315df71d/providers/Microsoft.HybridCompute/locations/eastus/operationstatus/372f7387-0f1a-4eb0-a62a-369c779c3cc7?api-version=2019-12-12.'

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
VERBOSE: INFORMATION: POLLING Parameter: ''
POLLING RequestMessage 'Method: PUT, RequestUri: 'https://management.azure.com/subscriptions/422b6c61-95b0-4213-b3be-7282315df71d/resourceGroups/arc-server-test/providers/Microsoft.HybridCompute/machines/tyler-ThinkPad-X1-Carbon/extensions/foo?api-version=2019-12-12', Version: 1.1, Content: System.Net.Http.StringContent, Headers:
{
  x-ms-unique-id: 27
  x-ms-client-request-id: c7c5b66b-b95f-4d23-900b-24a34d04b320
  CommandName: Set-AzConnectedMachineExtension
  FullCommandName: Set-AzConnectedMachineExtension_Update
  ParameterSetName: __AllParameterSets
  User-Agent: AzurePowershell/Az4.0.0-preview
  Authorization: Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsIng1dCI6ImppYk5ia0ZTU2JteFBZck45Q0ZxUms0SzRndyIsImtpZCI6ImppYk5ia0ZTU2JteFBZck45Q0ZxUms0SzRndyJ9.eyJhdWQiOiJodHRwczovL21hbmFnZW1lbnQuY29yZS53aW5kb3dzLm5ldC8iLCJpc3MiOiJodHRwczovL3N0cy53aW5kb3dzLm5ldC83MmY5ODhiZi04NmYxLTQxYWYtOTFhYi0yZDdjZDAxMWRiNDcvIiwiaWF0IjoxNTk4NDE4MTQ4LCJuYmYiOjE1OTg0MTgxNDgsImV4cCI6MTU5ODQyMjA0OCwiX2NsYWltX25hbWVzIjp7Imdyb3VwcyI6InNyYzEifSwiX2NsYWltX3NvdXJjZXMiOnsic3JjMSI6eyJlbmRwb2ludCI6Imh0dHBzOi8vZ3JhcGgud2luZG93cy5uZXQvNzJmOTg4YmYtODZmMS00MWFmLTkxYWItMmQ3Y2QwMTFkYjQ3L3VzZXJzL2MyNTZjN2IwLWU5NGQtNDljNC04NTg1LTBjN2YzMDIwYmQzNy9nZXRNZW1iZXJPYmplY3RzIn19LCJhY3IiOiIxIiwiYWlvIjoiQVZRQXEvOFFBQUFBcnNiRnYvMU5kdXVpcjJuNy9nYUkzYkw4SkVNY3E1YnVlOEtZMGE3bk5VK3YwOFpIT1YzTG5EZGxqWkxrZUZRbk52OEJ5NUlMZkpIY3ErN29vaEFLQmx1bUV4Z1NNdis2OXRPVTRWa3pOcFE9IiwiYW1yIjpbInB3ZCIsIm1mYSJdLCJhcHBpZCI6IjE5NTBhMjU4LTIyN2ItNGUzMS1hOWNmLTcxNzQ5NTk0NWZjMiIsImFwcGlkYWNyIjoiMCIsImZhbWlseV9uYW1lIjoiTGVvbmhhcmR0IChQT1dFUlNIRUxMKSIsImdpdmVuX25hbWUiOiJUeWxlciIsImlwYWRkciI6Ijc1LjE3Mi4xODMuMTc2IiwibmFtZSI6IlR5bGVyIExlb25oYXJkdCAoUE9XRVJTSEVMTCkiLCJvaWQiOiJjMjU2YzdiMC1lOTRkLTQ5YzQtODU4NS0wYzdmMzAyMGJkMzciLCJvbnByZW1fc2lkIjoiUy0xLTUtMjEtMjEyNzUyMTE4NC0xNjA0MDEyOTIwLTE4ODc5Mjc1MjctMjIyNjkzMzgiLCJwdWlkIjoiMTAwMzdGRkU5OThDRkZGNyIsInJoIjoiMC5BUm9BdjRqNWN2R0dyMEdScXkxODBCSGJSMWlpVUJsN0lqRk9xYzl4ZEpXVVg4SWFBTk0uIiwic2NwIjoidXNlcl9pbXBlcnNvbmF0aW9uIiwic3ViIjoiMXJrT05YeDAtUDVJQnlFc2kyV1RROVZXQUtmNW9IQjRVRnNkV040aFlBSSIsInRpZCI6IjcyZjk4OGJmLTg2ZjEtNDFhZi05MWFiLTJkN2NkMDExZGI0NyIsInVuaXF1ZV9uYW1lIjoidHlsZW9uaGFAbWljcm9zb2Z0LmNvbSIsInVwbiI6InR5bGVvbmhhQG1pY3Jvc29mdC5jb20iLCJ1dGkiOiJZYzF3TWtsUUYwMkRiU1k5RXR3TEFBIiwidmVyIjoiMS4wIiwieG1zX3RjZHQiOjEyODkyNDE1NDd9.VbmeQcgINWvA1hIQ2IOZh1ND1QpY0qwo5UQm21Y2zXTJ6ah1PEJN4iSWOxk8KWsc7lpNB3BBEWtgoT4YDhE79kmFZl-miiw2CRTstzrYLSUVD4vG_yAhjJ6b7q3bgv_fnvioZ6TekJ7J9DNQWTLC_IVaD57-vnN8JDbG-GAD356Zl_ywBKGhZoJkoJPK-O_zb73SpP6kUw8uxa49UGqI3qt7VVAVYRIkSnPucLZu5VjwEBSGxzVpuUANh7Ag7Pz8k1ZXEupjTcsF2kcba3vVumtHm90PmpXoJlkJDQV8os7MpDaRJOcbc1p5FlzpiReL70HXxumZB2H5HmbCbucX1A
  Content-Type: application/json
  Content-Length: 555
}'
POLLING Response: 'StatusCode: 202, ReasonPhrase: 'Accepted', Version: 1.1, Content: System.Net.Http.HttpConnectionResponseContent, Headers:
{
  Cache-Control: no-cache
  Pragma: no-cache
  Location: https://management.azure.com/subscriptions/422b6c61-95b0-4213-b3be-7282315df71d/providers/Microsoft.HybridCompute/locations/eastus/operationresults/372f7387-0f1a-4eb0-a62a-369c779c3cc7?api-version=2019-12-12
  Strict-Transport-Security: max-age=31536000; includeSubDomains
  x-ms-request-id: 7fc67a75-d2b1-4f2d-af48-364100a83ebc
  Azure-AsyncOperation: https://management.azure.com/subscriptions/422b6c61-95b0-4213-b3be-7282315df71d/providers/Microsoft.HybridCompute/locations/eastus/operationstatus/372f7387-0f1a-4eb0-a62a-369c779c3cc7?api-version=2019-12-12
  Server: Kestrel
  x-ms-ratelimit-remaining-subscription-resource-requests: 319
  x-ms-correlation-request-id: 883efb04-8272-4591-93a5-fba69fae8faf
  x-ms-routing-request-id: WESTUS2:20200826T051005Z:883efb04-8272-4591-93a5-fba69fae8faf
  X-Content-Type-Options: nosniff
  Date: Wed, 26 Aug 2020 05:10:04 GMT
  Content-Type: application/json
  Expires: -1
  Content-Length: 0
}'
POLLING Value: '0'

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
VERBOSE: INFORMATION: POLLING ExtendedData Type: ''
POLLING ExtendedData ''

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
DEBUG: Polling: Polling https://management.azure.com/subscriptions/422b6c61-95b0-4213-b3be-7282315df71d/providers/Microsoft.HybridCompute/locations/eastus/operationstatus/372f7387-0f1a-4eb0-a62a-369c779c3cc7?api-version=2019-12-12.
DEBUG: [ADAL]: Information: 2020-08-26T05:17:03.3011534Z: 461e0005-3284-4ea9-89dd-b0eb126c45d2 - LoggerBase.cs: ADAL PCL.CoreCLR with assembly version '3.19.2.6005', file version '3.19.50302.0130' and informational version '2a8bec6c4c76d0c1ef819b55bdc3cda2d2605056' is running...
DEBUG: [ADAL]: Information: 2020-08-26T05:17:03.3013201Z: 461e0005-3284-4ea9-89dd-b0eb126c45d2 - LoggerBase.cs: ADAL PCL.CoreCLR with assembly version '3.19.2.6005', file version '3.19.50302.0130' and informational version '2a8bec6c4c76d0c1ef819b55bdc3cda2d2605056' is running...
DEBUG: [ADAL]: Information: 2020-08-26T05:17:03.3014510Z: 461e0005-3284-4ea9-89dd-b0eb126c45d2 - LoggerBase.cs: === Token Acquisition started: 
	CacheType: null
	Authentication Target: User
	, Authority Host: login.microsoftonline.com
DEBUG: [ADAL]: Information: 2020-08-26T05:17:03.3015502Z: 461e0005-3284-4ea9-89dd-b0eb126c45d2 - LoggerBase.cs: === Token Acquisition started:
	Authority: https://login.microsoftonline.com/72f988bf-86f1-41af-91ab-2d7cd011db47/
	Resource: https://management.core.windows.net/
	ClientId: 1950a258-227b-4e31-a9cf-717495945fc2
	CacheType: null
	Authentication Target: User
	
DEBUG: [ADAL]: Verbose: 2020-08-26T05:17:03.3017349Z: 461e0005-3284-4ea9-89dd-b0eb126c45d2 - LoggerBase.cs: Loading from cache.
DEBUG: [ADAL]: Verbose: 2020-08-26T05:17:03.3018153Z: 461e0005-3284-4ea9-89dd-b0eb126c45d2 - LoggerBase.cs: Loading from cache.
DEBUG: [ADAL]: Information: 2020-08-26T05:17:03.3031761Z: 00000000-0000-0000-0000-000000000000 - LoggerBase.cs: Deserialized 3 items to token cache.
DEBUG: [ADAL]: Verbose: 2020-08-26T05:17:03.3032263Z: 461e0005-3284-4ea9-89dd-b0eb126c45d2 - LoggerBase.cs: Looking up cache for a token...
DEBUG: [ADAL]: Verbose: 2020-08-26T05:17:03.3032465Z: 461e0005-3284-4ea9-89dd-b0eb126c45d2 - LoggerBase.cs: Looking up cache for a token...
DEBUG: [ADAL]: Information: 2020-08-26T05:17:03.3032812Z: 461e0005-3284-4ea9-89dd-b0eb126c45d2 - LoggerBase.cs: An item matching the requested resource was found in the cache
DEBUG: [ADAL]: Information: 2020-08-26T05:17:03.3032992Z: 461e0005-3284-4ea9-89dd-b0eb126c45d2 - LoggerBase.cs: An item matching the requested resource was found in the cache
DEBUG: [ADAL]: Information: 2020-08-26T05:17:03.3033287Z: 461e0005-3284-4ea9-89dd-b0eb126c45d2 - LoggerBase.cs: 50.381527965 minutes left until token in cache expires
DEBUG: [ADAL]: Information: 2020-08-26T05:17:03.3033656Z: 461e0005-3284-4ea9-89dd-b0eb126c45d2 - LoggerBase.cs: 50.381527965 minutes left until token in cache expires
DEBUG: [ADAL]: Information: 2020-08-26T05:17:03.3034187Z: 461e0005-3284-4ea9-89dd-b0eb126c45d2 - LoggerBase.cs: A matching item (access token or refresh token or both) was found in the cache
DEBUG: [ADAL]: Information: 2020-08-26T05:17:03.3034769Z: 461e0005-3284-4ea9-89dd-b0eb126c45d2 - LoggerBase.cs: A matching item (access token or refresh token or both) was found in the cache
DEBUG: [ADAL]: Information: 2020-08-26T05:17:03.3036716Z: 461e0005-3284-4ea9-89dd-b0eb126c45d2 - LoggerBase.cs: === Token Acquisition finished successfully. An access token was returned: Expiration Time: 8/26/2020 6:07:26 AM +00:00
DEBUG: [ADAL]: Information: 2020-08-26T05:17:03.3037473Z: 461e0005-3284-4ea9-89dd-b0eb126c45d2 - LoggerBase.cs: === Token Acquisition finished successfully. An access token was returned: Expiration Time: 8/26/2020 6:07:26 AM +00:00Access Token Hash: XNvbFj1C+pyfghSu12Vz3fD8JfvYsZTHp4Ff75DRkDA=
	 User id: c256c7b0-e94d-49c4-8585-0c7f3020bd37
VERBOSE: INFORMATION: POLLING The contents are 'Polling' and 'Polled https://management.azure.com/subscriptions/422b6c61-95b0-4213-b3be-7282315df71d/providers/Microsoft.HybridCompute/locations/eastus/operationstatus/372f7387-0f1a-4eb0-a62a-369c779c3cc7?api-version=2019-12-12 provisioning state  Failed.'

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
VERBOSE: INFORMATION: POLLING Parameter: ''
POLLING RequestMessage 'Method: GET, RequestUri: 'https://management.azure.com/subscriptions/422b6c61-95b0-4213-b3be-7282315df71d/providers/Microsoft.HybridCompute/locations/eastus/operationstatus/372f7387-0f1a-4eb0-a62a-369c779c3cc7?api-version=2019-12-12', Version: 1.1, Content: <null>, Headers:
{
  x-ms-unique-id: 27
  x-ms-unique-id: 28
  x-ms-client-request-id: c7c5b66b-b95f-4d23-900b-24a34d04b320
  x-ms-client-request-id: c7c5b66b-b95f-4d23-900b-24a34d04b320
  CommandName: Set-AzConnectedMachineExtension
  CommandName: Set-AzConnectedMachineExtension
  FullCommandName: Set-AzConnectedMachineExtension_Update
  FullCommandName: Set-AzConnectedMachineExtension_Update
  ParameterSetName: __AllParameterSets
  ParameterSetName: __AllParameterSets
  User-Agent: AzurePowershell/Az4.0.0-preview
  User-Agent: AzurePowershell/Az4.0.0-preview
  Authorization: Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsIng1dCI6ImppYk5ia0ZTU2JteFBZck45Q0ZxUms0SzRndyIsImtpZCI6ImppYk5ia0ZTU2JteFBZck45Q0ZxUms0SzRndyJ9.eyJhdWQiOiJodHRwczovL21hbmFnZW1lbnQuY29yZS53aW5kb3dzLm5ldC8iLCJpc3MiOiJodHRwczovL3N0cy53aW5kb3dzLm5ldC83MmY5ODhiZi04NmYxLTQxYWYtOTFhYi0yZDdjZDAxMWRiNDcvIiwiaWF0IjoxNTk4NDE4MTQ4LCJuYmYiOjE1OTg0MTgxNDgsImV4cCI6MTU5ODQyMjA0OCwiX2NsYWltX25hbWVzIjp7Imdyb3VwcyI6InNyYzEifSwiX2NsYWltX3NvdXJjZXMiOnsic3JjMSI6eyJlbmRwb2ludCI6Imh0dHBzOi8vZ3JhcGgud2luZG93cy5uZXQvNzJmOTg4YmYtODZmMS00MWFmLTkxYWItMmQ3Y2QwMTFkYjQ3L3VzZXJzL2MyNTZjN2IwLWU5NGQtNDljNC04NTg1LTBjN2YzMDIwYmQzNy9nZXRNZW1iZXJPYmplY3RzIn19LCJhY3IiOiIxIiwiYWlvIjoiQVZRQXEvOFFBQUFBcnNiRnYvMU5kdXVpcjJuNy9nYUkzYkw4SkVNY3E1YnVlOEtZMGE3bk5VK3YwOFpIT1YzTG5EZGxqWkxrZUZRbk52OEJ5NUlMZkpIY3ErN29vaEFLQmx1bUV4Z1NNdis2OXRPVTRWa3pOcFE9IiwiYW1yIjpbInB3ZCIsIm1mYSJdLCJhcHBpZCI6IjE5NTBhMjU4LTIyN2ItNGUzMS1hOWNmLTcxNzQ5NTk0NWZjMiIsImFwcGlkYWNyIjoiMCIsImZhbWlseV9uYW1lIjoiTGVvbmhhcmR0IChQT1dFUlNIRUxMKSIsImdpdmVuX25hbWUiOiJUeWxlciIsImlwYWRkciI6Ijc1LjE3Mi4xODMuMTc2IiwibmFtZSI6IlR5bGVyIExlb25oYXJkdCAoUE9XRVJTSEVMTCkiLCJvaWQiOiJjMjU2YzdiMC1lOTRkLTQ5YzQtODU4NS0wYzdmMzAyMGJkMzciLCJvbnByZW1fc2lkIjoiUy0xLTUtMjEtMjEyNzUyMTE4NC0xNjA0MDEyOTIwLTE4ODc5Mjc1MjctMjIyNjkzMzgiLCJwdWlkIjoiMTAwMzdGRkU5OThDRkZGNyIsInJoIjoiMC5BUm9BdjRqNWN2R0dyMEdScXkxODBCSGJSMWlpVUJsN0lqRk9xYzl4ZEpXVVg4SWFBTk0uIiwic2NwIjoidXNlcl9pbXBlcnNvbmF0aW9uIiwic3ViIjoiMXJrT05YeDAtUDVJQnlFc2kyV1RROVZXQUtmNW9IQjRVRnNkV040aFlBSSIsInRpZCI6IjcyZjk4OGJmLTg2ZjEtNDFhZi05MWFiLTJkN2NkMDExZGI0NyIsInVuaXF1ZV9uYW1lIjoidHlsZW9uaGFAbWljcm9zb2Z0LmNvbSIsInVwbiI6InR5bGVvbmhhQG1pY3Jvc29mdC5jb20iLCJ1dGkiOiJZYzF3TWtsUUYwMkRiU1k5RXR3TEFBIiwidmVyIjoiMS4wIiwieG1zX3RjZHQiOjEyODkyNDE1NDd9.VbmeQcgINWvA1hIQ2IOZh1ND1QpY0qwo5UQm21Y2zXTJ6ah1PEJN4iSWOxk8KWsc7lpNB3BBEWtgoT4YDhE79kmFZl-miiw2CRTstzrYLSUVD4vG_yAhjJ6b7q3bgv_fnvioZ6TekJ7J9DNQWTLC_IVaD57-vnN8JDbG-GAD356Zl_ywBKGhZoJkoJPK-O_zb73SpP6kUw8uxa49UGqI3qt7VVAVYRIkSnPucLZu5VjwEBSGxzVpuUANh7Ag7Pz8k1ZXEupjTcsF2kcba3vVumtHm90PmpXoJlkJDQV8os7MpDaRJOcbc1p5FlzpiReL70HXxumZB2H5HmbCbucX1A
}'
POLLING Response: 'StatusCode: 200, ReasonPhrase: 'OK', Version: 1.1, Content: System.Net.Http.HttpConnectionResponseContent, Headers:
{
  Cache-Control: no-cache
  Pragma: no-cache
  x-ms-ratelimit-remaining-subscription-reads: 11999
  Strict-Transport-Security: max-age=31536000; includeSubDomains
  x-ms-request-id: 1584595a-d876-40ef-bdad-b8726c1bd99b
  Server: Kestrel
  x-ms-correlation-request-id: a253dafa-d797-4622-9d22-4a895e4bde37
  x-ms-routing-request-id: WESTUS2:20200826T051705Z:a253dafa-d797-4622-9d22-4a895e4bde37
  X-Content-Type-Options: nosniff
  Date: Wed, 26 Aug 2020 05:17:04 GMT
  Content-Length: 159
  Content-Type: application/json; charset=utf-8
  Expires: -1
}'
POLLING Value: '0'

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
VERBOSE: INFORMATION: POLLING ExtendedData Type: ''
POLLING ExtendedData ''

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
DEBUG: Polling: Polled https://management.azure.com/subscriptions/422b6c61-95b0-4213-b3be-7282315df71d/providers/Microsoft.HybridCompute/locations/eastus/operationstatus/372f7387-0f1a-4eb0-a62a-369c779c3cc7?api-version=2019-12-12 provisioning state  Failed.
DEBUG: [ADAL]: Information: 2020-08-26T05:17:19.3284484Z: f9422e70-c545-498c-a93c-a611d60f7493 - LoggerBase.cs: ADAL PCL.CoreCLR with assembly version '3.19.2.6005', file version '3.19.50302.0130' and informational version '2a8bec6c4c76d0c1ef819b55bdc3cda2d2605056' is running...
DEBUG: [ADAL]: Information: 2020-08-26T05:17:19.3286810Z: f9422e70-c545-498c-a93c-a611d60f7493 - LoggerBase.cs: ADAL PCL.CoreCLR with assembly version '3.19.2.6005', file version '3.19.50302.0130' and informational version '2a8bec6c4c76d0c1ef819b55bdc3cda2d2605056' is running...
DEBUG: [ADAL]: Information: 2020-08-26T05:17:19.3288494Z: f9422e70-c545-498c-a93c-a611d60f7493 - LoggerBase.cs: === Token Acquisition started: 
	CacheType: null
	Authentication Target: User
	, Authority Host: login.microsoftonline.com
DEBUG: [ADAL]: Information: 2020-08-26T05:17:19.3289485Z: f9422e70-c545-498c-a93c-a611d60f7493 - LoggerBase.cs: === Token Acquisition started:
	Authority: https://login.microsoftonline.com/72f988bf-86f1-41af-91ab-2d7cd011db47/
	Resource: https://management.core.windows.net/
	ClientId: 1950a258-227b-4e31-a9cf-717495945fc2
	CacheType: null
	Authentication Target: User
	
DEBUG: [ADAL]: Verbose: 2020-08-26T05:17:19.3295885Z: f9422e70-c545-498c-a93c-a611d60f7493 - LoggerBase.cs: Loading from cache.
DEBUG: [ADAL]: Verbose: 2020-08-26T05:17:19.3296732Z: f9422e70-c545-498c-a93c-a611d60f7493 - LoggerBase.cs: Loading from cache.
DEBUG: [ADAL]: Information: 2020-08-26T05:17:19.3311474Z: 00000000-0000-0000-0000-000000000000 - LoggerBase.cs: Deserialized 3 items to token cache.
DEBUG: [ADAL]: Verbose: 2020-08-26T05:17:19.3312096Z: f9422e70-c545-498c-a93c-a611d60f7493 - LoggerBase.cs: Looking up cache for a token...
DEBUG: [ADAL]: Verbose: 2020-08-26T05:17:19.3312321Z: f9422e70-c545-498c-a93c-a611d60f7493 - LoggerBase.cs: Looking up cache for a token...
DEBUG: [ADAL]: Information: 2020-08-26T05:17:19.3312659Z: f9422e70-c545-498c-a93c-a611d60f7493 - LoggerBase.cs: An item matching the requested resource was found in the cache
DEBUG: [ADAL]: Information: 2020-08-26T05:17:19.3312858Z: f9422e70-c545-498c-a93c-a611d60f7493 - LoggerBase.cs: An item matching the requested resource was found in the cache
DEBUG: [ADAL]: Information: 2020-08-26T05:17:19.3313274Z: f9422e70-c545-498c-a93c-a611d60f7493 - LoggerBase.cs: 50.114394665 minutes left until token in cache expires
DEBUG: [ADAL]: Information: 2020-08-26T05:17:19.3313471Z: f9422e70-c545-498c-a93c-a611d60f7493 - LoggerBase.cs: 50.114394665 minutes left until token in cache expires
DEBUG: [ADAL]: Information: 2020-08-26T05:17:19.3313639Z: f9422e70-c545-498c-a93c-a611d60f7493 - LoggerBase.cs: A matching item (access token or refresh token or both) was found in the cache
DEBUG: [ADAL]: Information: 2020-08-26T05:17:19.3313824Z: f9422e70-c545-498c-a93c-a611d60f7493 - LoggerBase.cs: A matching item (access token or refresh token or both) was found in the cache
DEBUG: [ADAL]: Information: 2020-08-26T05:17:19.3314566Z: f9422e70-c545-498c-a93c-a611d60f7493 - LoggerBase.cs: === Token Acquisition finished successfully. An access token was returned: Expiration Time: 8/26/2020 6:07:26 AM +00:00
DEBUG: [ADAL]: Information: 2020-08-26T05:17:19.3314810Z: f9422e70-c545-498c-a93c-a611d60f7493 - LoggerBase.cs: === Token Acquisition finished successfully. An access token was returned: Expiration Time: 8/26/2020 6:07:26 AM +00:00Access Token Hash: XNvbFj1C+pyfghSu12Vz3fD8JfvYsZTHp4Ff75DRkDA=
	 User id: c256c7b0-e94d-49c4-8585-0c7f3020bd37
VERBOSE: INFORMATION: BEFORE RESPONSE DISPATCH The contents are 'BeforeResponseDispatch' and ''

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
VERBOSE: INFORMATION: BEFORE RESPONSE DISPATCH Parameter: ''
BEFORE RESPONSE DISPATCH RequestMessage 'Method: GET, RequestUri: 'https://management.azure.com/subscriptions/422b6c61-95b0-4213-b3be-7282315df71d/resourceGroups/arc-server-test/providers/Microsoft.HybridCompute/machines/tyler-ThinkPad-X1-Carbon/extensions/foo?api-version=2019-12-12', Version: 1.1, Content: <null>, Headers:
{
  x-ms-unique-id: 27
  x-ms-unique-id: 28
  x-ms-unique-id: 29
  x-ms-client-request-id: c7c5b66b-b95f-4d23-900b-24a34d04b320
  x-ms-client-request-id: c7c5b66b-b95f-4d23-900b-24a34d04b320
  x-ms-client-request-id: c7c5b66b-b95f-4d23-900b-24a34d04b320
  CommandName: Set-AzConnectedMachineExtension
  CommandName: Set-AzConnectedMachineExtension
  CommandName: Set-AzConnectedMachineExtension
  FullCommandName: Set-AzConnectedMachineExtension_Update
  FullCommandName: Set-AzConnectedMachineExtension_Update
  FullCommandName: Set-AzConnectedMachineExtension_Update
  ParameterSetName: __AllParameterSets
  ParameterSetName: __AllParameterSets
  ParameterSetName: __AllParameterSets
  User-Agent: AzurePowershell/Az4.0.0-preview
  User-Agent: AzurePowershell/Az4.0.0-preview
  User-Agent: AzurePowershell/Az4.0.0-preview
  Authorization: Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsIng1dCI6ImppYk5ia0ZTU2JteFBZck45Q0ZxUms0SzRndyIsImtpZCI6ImppYk5ia0ZTU2JteFBZck45Q0ZxUms0SzRndyJ9.eyJhdWQiOiJodHRwczovL21hbmFnZW1lbnQuY29yZS53aW5kb3dzLm5ldC8iLCJpc3MiOiJodHRwczovL3N0cy53aW5kb3dzLm5ldC83MmY5ODhiZi04NmYxLTQxYWYtOTFhYi0yZDdjZDAxMWRiNDcvIiwiaWF0IjoxNTk4NDE4MTQ4LCJuYmYiOjE1OTg0MTgxNDgsImV4cCI6MTU5ODQyMjA0OCwiX2NsYWltX25hbWVzIjp7Imdyb3VwcyI6InNyYzEifSwiX2NsYWltX3NvdXJjZXMiOnsic3JjMSI6eyJlbmRwb2ludCI6Imh0dHBzOi8vZ3JhcGgud2luZG93cy5uZXQvNzJmOTg4YmYtODZmMS00MWFmLTkxYWItMmQ3Y2QwMTFkYjQ3L3VzZXJzL2MyNTZjN2IwLWU5NGQtNDljNC04NTg1LTBjN2YzMDIwYmQzNy9nZXRNZW1iZXJPYmplY3RzIn19LCJhY3IiOiIxIiwiYWlvIjoiQVZRQXEvOFFBQUFBcnNiRnYvMU5kdXVpcjJuNy9nYUkzYkw4SkVNY3E1YnVlOEtZMGE3bk5VK3YwOFpIT1YzTG5EZGxqWkxrZUZRbk52OEJ5NUlMZkpIY3ErN29vaEFLQmx1bUV4Z1NNdis2OXRPVTRWa3pOcFE9IiwiYW1yIjpbInB3ZCIsIm1mYSJdLCJhcHBpZCI6IjE5NTBhMjU4LTIyN2ItNGUzMS1hOWNmLTcxNzQ5NTk0NWZjMiIsImFwcGlkYWNyIjoiMCIsImZhbWlseV9uYW1lIjoiTGVvbmhhcmR0IChQT1dFUlNIRUxMKSIsImdpdmVuX25hbWUiOiJUeWxlciIsImlwYWRkciI6Ijc1LjE3Mi4xODMuMTc2IiwibmFtZSI6IlR5bGVyIExlb25oYXJkdCAoUE9XRVJTSEVMTCkiLCJvaWQiOiJjMjU2YzdiMC1lOTRkLTQ5YzQtODU4NS0wYzdmMzAyMGJkMzciLCJvbnByZW1fc2lkIjoiUy0xLTUtMjEtMjEyNzUyMTE4NC0xNjA0MDEyOTIwLTE4ODc5Mjc1MjctMjIyNjkzMzgiLCJwdWlkIjoiMTAwMzdGRkU5OThDRkZGNyIsInJoIjoiMC5BUm9BdjRqNWN2R0dyMEdScXkxODBCSGJSMWlpVUJsN0lqRk9xYzl4ZEpXVVg4SWFBTk0uIiwic2NwIjoidXNlcl9pbXBlcnNvbmF0aW9uIiwic3ViIjoiMXJrT05YeDAtUDVJQnlFc2kyV1RROVZXQUtmNW9IQjRVRnNkV040aFlBSSIsInRpZCI6IjcyZjk4OGJmLTg2ZjEtNDFhZi05MWFiLTJkN2NkMDExZGI0NyIsInVuaXF1ZV9uYW1lIjoidHlsZW9uaGFAbWljcm9zb2Z0LmNvbSIsInVwbiI6InR5bGVvbmhhQG1pY3Jvc29mdC5jb20iLCJ1dGkiOiJZYzF3TWtsUUYwMkRiU1k5RXR3TEFBIiwidmVyIjoiMS4wIiwieG1zX3RjZHQiOjEyODkyNDE1NDd9.VbmeQcgINWvA1hIQ2IOZh1ND1QpY0qwo5UQm21Y2zXTJ6ah1PEJN4iSWOxk8KWsc7lpNB3BBEWtgoT4YDhE79kmFZl-miiw2CRTstzrYLSUVD4vG_yAhjJ6b7q3bgv_fnvioZ6TekJ7J9DNQWTLC_IVaD57-vnN8JDbG-GAD356Zl_ywBKGhZoJkoJPK-O_zb73SpP6kUw8uxa49UGqI3qt7VVAVYRIkSnPucLZu5VjwEBSGxzVpuUANh7Ag7Pz8k1ZXEupjTcsF2kcba3vVumtHm90PmpXoJlkJDQV8os7MpDaRJOcbc1p5FlzpiReL70HXxumZB2H5HmbCbucX1A
}'
BEFORE RESPONSE DISPATCH Response: 'StatusCode: 200, ReasonPhrase: 'OK', Version: 1.1, Content: System.Net.Http.HttpConnectionResponseContent, Headers:
{
  Cache-Control: no-cache
  Pragma: no-cache
  x-ms-ratelimit-remaining-subscription-reads: 11998
  Strict-Transport-Security: max-age=31536000; includeSubDomains
  x-ms-request-id: f68bd6e3-a137-400f-9252-e8574d94e34e
  Server: Kestrel
  x-ms-correlation-request-id: 418385f3-cea0-47b2-9dc6-82150a117b85
  x-ms-routing-request-id: WESTUS2:20200826T051719Z:418385f3-cea0-47b2-9dc6-82150a117b85
  X-Content-Type-Options: nosniff
  Date: Wed, 26 Aug 2020 05:17:19 GMT
  Content-Length: 688
  Content-Type: application/json; charset=utf-8
  Expires: -1
}'
BEFORE RESPONSE DISPATCH Value: '0'

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
VERBOSE: INFORMATION: BEFORE RESPONSE DISPATCH ExtendedData Type: ''
BEFORE RESPONSE DISPATCH ExtendedData ''

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
DEBUG: BeforeResponseDispatch: 

VERBOSE: INFORMATION: FINALLY The contents are 'Finally' and ''

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
VERBOSE: INFORMATION: FINALLY Parameter: ''
FINALLY RequestMessage 'Method: GET, RequestUri: 'https://management.azure.com/subscriptions/422b6c61-95b0-4213-b3be-7282315df71d/resourceGroups/arc-server-test/providers/Microsoft.HybridCompute/machines/tyler-ThinkPad-X1-Carbon/extensions/foo?api-version=2019-12-12', Version: 1.1, Content: <null>, Headers:
{
  x-ms-unique-id: 27
  x-ms-unique-id: 28
  x-ms-unique-id: 29
  x-ms-client-request-id: c7c5b66b-b95f-4d23-900b-24a34d04b320
  x-ms-client-request-id: c7c5b66b-b95f-4d23-900b-24a34d04b320
  x-ms-client-request-id: c7c5b66b-b95f-4d23-900b-24a34d04b320
  CommandName: Set-AzConnectedMachineExtension
  CommandName: Set-AzConnectedMachineExtension
  CommandName: Set-AzConnectedMachineExtension
  FullCommandName: Set-AzConnectedMachineExtension_Update
  FullCommandName: Set-AzConnectedMachineExtension_Update
  FullCommandName: Set-AzConnectedMachineExtension_Update
  ParameterSetName: __AllParameterSets
  ParameterSetName: __AllParameterSets
  ParameterSetName: __AllParameterSets
  User-Agent: AzurePowershell/Az4.0.0-preview
  User-Agent: AzurePowershell/Az4.0.0-preview
  User-Agent: AzurePowershell/Az4.0.0-preview
  Authorization: Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsIng1dCI6ImppYk5ia0ZTU2JteFBZck45Q0ZxUms0SzRndyIsImtpZCI6ImppYk5ia0ZTU2JteFBZck45Q0ZxUms0SzRndyJ9.eyJhdWQiOiJodHRwczovL21hbmFnZW1lbnQuY29yZS53aW5kb3dzLm5ldC8iLCJpc3MiOiJodHRwczovL3N0cy53aW5kb3dzLm5ldC83MmY5ODhiZi04NmYxLTQxYWYtOTFhYi0yZDdjZDAxMWRiNDcvIiwiaWF0IjoxNTk4NDE4MTQ4LCJuYmYiOjE1OTg0MTgxNDgsImV4cCI6MTU5ODQyMjA0OCwiX2NsYWltX25hbWVzIjp7Imdyb3VwcyI6InNyYzEifSwiX2NsYWltX3NvdXJjZXMiOnsic3JjMSI6eyJlbmRwb2ludCI6Imh0dHBzOi8vZ3JhcGgud2luZG93cy5uZXQvNzJmOTg4YmYtODZmMS00MWFmLTkxYWItMmQ3Y2QwMTFkYjQ3L3VzZXJzL2MyNTZjN2IwLWU5NGQtNDljNC04NTg1LTBjN2YzMDIwYmQzNy9nZXRNZW1iZXJPYmplY3RzIn19LCJhY3IiOiIxIiwiYWlvIjoiQVZRQXEvOFFBQUFBcnNiRnYvMU5kdXVpcjJuNy9nYUkzYkw4SkVNY3E1YnVlOEtZMGE3bk5VK3YwOFpIT1YzTG5EZGxqWkxrZUZRbk52OEJ5NUlMZkpIY3ErN29vaEFLQmx1bUV4Z1NNdis2OXRPVTRWa3pOcFE9IiwiYW1yIjpbInB3ZCIsIm1mYSJdLCJhcHBpZCI6IjE5NTBhMjU4LTIyN2ItNGUzMS1hOWNmLTcxNzQ5NTk0NWZjMiIsImFwcGlkYWNyIjoiMCIsImZhbWlseV9uYW1lIjoiTGVvbmhhcmR0IChQT1dFUlNIRUxMKSIsImdpdmVuX25hbWUiOiJUeWxlciIsImlwYWRkciI6Ijc1LjE3Mi4xODMuMTc2IiwibmFtZSI6IlR5bGVyIExlb25oYXJkdCAoUE9XRVJTSEVMTCkiLCJvaWQiOiJjMjU2YzdiMC1lOTRkLTQ5YzQtODU4NS0wYzdmMzAyMGJkMzciLCJvbnByZW1fc2lkIjoiUy0xLTUtMjEtMjEyNzUyMTE4NC0xNjA0MDEyOTIwLTE4ODc5Mjc1MjctMjIyNjkzMzgiLCJwdWlkIjoiMTAwMzdGRkU5OThDRkZGNyIsInJoIjoiMC5BUm9BdjRqNWN2R0dyMEdScXkxODBCSGJSMWlpVUJsN0lqRk9xYzl4ZEpXVVg4SWFBTk0uIiwic2NwIjoidXNlcl9pbXBlcnNvbmF0aW9uIiwic3ViIjoiMXJrT05YeDAtUDVJQnlFc2kyV1RROVZXQUtmNW9IQjRVRnNkV040aFlBSSIsInRpZCI6IjcyZjk4OGJmLTg2ZjEtNDFhZi05MWFiLTJkN2NkMDExZGI0NyIsInVuaXF1ZV9uYW1lIjoidHlsZW9uaGFAbWljcm9zb2Z0LmNvbSIsInVwbiI6InR5bGVvbmhhQG1pY3Jvc29mdC5jb20iLCJ1dGkiOiJZYzF3TWtsUUYwMkRiU1k5RXR3TEFBIiwidmVyIjoiMS4wIiwieG1zX3RjZHQiOjEyODkyNDE1NDd9.VbmeQcgINWvA1hIQ2IOZh1ND1QpY0qwo5UQm21Y2zXTJ6ah1PEJN4iSWOxk8KWsc7lpNB3BBEWtgoT4YDhE79kmFZl-miiw2CRTstzrYLSUVD4vG_yAhjJ6b7q3bgv_fnvioZ6TekJ7J9DNQWTLC_IVaD57-vnN8JDbG-GAD356Zl_ywBKGhZoJkoJPK-O_zb73SpP6kUw8uxa49UGqI3qt7VVAVYRIkSnPucLZu5VjwEBSGxzVpuUANh7Ag7Pz8k1ZXEupjTcsF2kcba3vVumtHm90PmpXoJlkJDQV8os7MpDaRJOcbc1p5FlzpiReL70HXxumZB2H5HmbCbucX1A
}'
FINALLY Response: 'StatusCode: 200, ReasonPhrase: 'OK', Version: 1.1, Content: System.Net.Http.HttpConnectionResponseContent, Headers:
{
  Cache-Control: no-cache
  Pragma: no-cache
  x-ms-ratelimit-remaining-subscription-reads: 11998
  Strict-Transport-Security: max-age=31536000; includeSubDomains
  x-ms-request-id: f68bd6e3-a137-400f-9252-e8574d94e34e
  Server: Kestrel
  x-ms-correlation-request-id: 418385f3-cea0-47b2-9dc6-82150a117b85
  x-ms-routing-request-id: WESTUS2:20200826T051719Z:418385f3-cea0-47b2-9dc6-82150a117b85
  X-Content-Type-Options: nosniff
  Date: Wed, 26 Aug 2020 05:17:19 GMT
  Content-Length: 688
  Content-Type: application/json; charset=utf-8
  Expires: -1
}'
FINALLY Value: '0'

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
VERBOSE: INFORMATION: FINALLY ExtendedData Type: ''
FINALLY ExtendedData ''

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
DEBUG: Finally: 
VERBOSE: INFORMATION: CMDLET AFTER API CALL The contents are 'CmdletAfterAPICall' and ''

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
VERBOSE: INFORMATION: CMDLET AFTER API CALL Parameter: ''
CMDLET AFTER API CALL RequestMessage ''
CMDLET AFTER API CALL Response: ''
CMDLET AFTER API CALL Value: '0'

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
VERBOSE: INFORMATION: CMDLET AFTER API CALL ExtendedData Type: ''
CMDLET AFTER API CALL ExtendedData ''

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
DEBUG: CmdletAfterAPICall: 
DEBUG: [CmdletProcessRecordAsyncEnd]: Sending new QosEvent for command 'Set-AzConnectedMachineExtension_Update': AzureQoSEvent: CommandName - Set-AzConnectedMachineExtension_Update; IsSuccess - True; Duration - 00:00:00;
DEBUG: CmdletProcessRecordAsyncEnd: 
VERBOSE: INFORMATION: CMDLET PROCESS RECORD END The contents are 'CmdletProcessRecordEnd' and ''

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
VERBOSE: INFORMATION: CMDLET PROCESS RECORD END Parameter: ''
CMDLET PROCESS RECORD END RequestMessage ''
CMDLET PROCESS RECORD END Response: ''
CMDLET PROCESS RECORD END Value: '0'

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
VERBOSE: INFORMATION: CMDLET PROCESS RECORD END ExtendedData Type: ''
CMDLET PROCESS RECORD END ExtendedData ''

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
DEBUG: CmdletProcessRecordEnd: 
VERBOSE: INFORMATION: CMDLET END PROCESSING The contents are 'CmdletEndProcessing' and ''

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
VERBOSE: INFORMATION: CMDLET END PROCESSING Parameter: ''
CMDLET END PROCESSING RequestMessage ''
CMDLET END PROCESSING Response: ''
CMDLET END PROCESSING Value: '0'

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
VERBOSE: INFORMATION: CMDLET END PROCESSING ExtendedData Type: ''
CMDLET END PROCESSING ExtendedData ''

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
DEBUG: CmdletEndProcessing: 

Name Location ProvisioningState
---- -------- -----------------
foo  eastus   Succeeded
```

Wanted to paste this output before I head to bed over here in Seattle. Let me know what you think. I can look closer from my perspective in the morning if you don't see anything unusual.

Reply:
I don't see anything out of the ordinary... although I can't see the result:

```
$a | Set-AzConnectedMachineExtension -MachineName (hostname) -ResourceGroupName arc-server-test -Name foo -Debug
DEBUG: 10:08:14 PM - GetAzureRMContextCommand begin processing with ParameterSet 'GetSingleContext'.
DEBUG: 10:08:14 PM - using account id 'tyleonha@microsoft.com'...
DEBUG: AzureQoSEvent: CommandName - Get-AzContext; IsSuccess - True; Duration - 00:00:00.0020808;
DEBUG: Finish sending metric.
DEBUG: 10:08:15 PM - GetAzureRMContextCommand end processing.
VERBOSE: INFORMATION: CMDLET BEGIN PROCESSING The contents are 'CmdletBeginProcessing' and ''

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): y
VERBOSE: INFORMATION: CMDLET BEGIN PROCESSING Parameter: ''
CMDLET BEGIN PROCESSING RequestMessage ''
CMDLET BEGIN PROCESSING Response: ''
CMDLET BEGIN PROCESSING Value: '0'

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
VERBOSE: INFORMATION: CMDLET BEGIN PROCESSING ExtendedData Type: ''
CMDLET BEGIN PROCESSING ExtendedData ''

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
DEBUG: CmdletBeginProcessing: 
VERBOSE: INFORMATION: CMDLET PROCESS RECORD START The contents are 'CmdletProcessRecordStart' and ''

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
VERBOSE: INFORMATION: CMDLET PROCESS RECORD START Parameter: ''
CMDLET PROCESS RECORD START RequestMessage ''
CMDLET PROCESS RECORD START Response: ''
CMDLET PROCESS RECORD START Value: '0'

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
VERBOSE: INFORMATION: CMDLET PROCESS RECORD START ExtendedData Type: ''
CMDLET PROCESS RECORD START ExtendedData ''

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
DEBUG: CmdletProcessRecordStart: 

Confirm
Are you sure you want to perform this action?
Performing the operation "Set-AzConnectedMachineExtension_Update" on target "Call remote 'MachineExtensionsCreateOrUpdate' operation".
[Y] Yes  [A] Yes to All  [N] No  [L] No to All  [S] Suspend  [?] Help (default is "Y"): 
DEBUG: [CmdletProcessRecordAsyncStart]: Created new QosEvent for command 'Set-AzConnectedMachineExtension_Update': AzureQoSEvent: CommandName - Set-AzConnectedMachineExtension_Update; IsSuccess - False; Duration - 00:00:00;
DEBUG: CmdletProcessRecordAsyncStart: 
VERBOSE: INFORMATION: CMDLET GET PIPELINE The contents are 'CmdletGetPipeline' and ''

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
VERBOSE: INFORMATION: CMDLET GET PIPELINE Parameter: ''
CMDLET GET PIPELINE RequestMessage ''
CMDLET GET PIPELINE Response: ''
CMDLET GET PIPELINE Value: '0'

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
VERBOSE: INFORMATION: CMDLET GET PIPELINE ExtendedData Type: ''
CMDLET GET PIPELINE ExtendedData ''

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
DEBUG: CmdletGetPipeline: 
VERBOSE: INFORMATION: CMDLET BEFORE API CALL The contents are 'CmdletBeforeAPICall' and ''

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
VERBOSE: INFORMATION: CMDLET BEFORE API CALL Parameter: ''
CMDLET BEFORE API CALL RequestMessage ''
CMDLET BEFORE API CALL Response: ''
CMDLET BEFORE API CALL Value: '0'

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
VERBOSE: INFORMATION: CMDLET BEFORE API CALL ExtendedData Type: ''
CMDLET BEFORE API CALL ExtendedData ''

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
DEBUG: CmdletBeforeAPICall: 
VERBOSE: INFORMATION: URL CREATED The contents are 'URLCreated' and 'https://management.azure.com/subscriptions/422b6c61-95b0-4213-b3be-7282315df71d/resourceGroups/arc-server-test/providers/Microsoft.HybridCompute/machines/tyler-ThinkPad-X1-Carbon/extensions/foo?api-version=2019-12-12'

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
VERBOSE: INFORMATION: URL CREATED Parameter: ''
URL CREATED RequestMessage ''
URL CREATED Response: ''
URL CREATED Value: '0'

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
VERBOSE: INFORMATION: URL CREATED ExtendedData Type: ''
URL CREATED ExtendedData ''

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
DEBUG: URLCreated: https://management.azure.com/subscriptions/422b6c61-95b0-4213-b3be-7282315df71d/resourceGroups/arc-server-test/providers/Microsoft.HybridCompute/machines/tyler-ThinkPad-X1-Carbon/extensions/foo?api-version=2019-12-12
VERBOSE: INFORMATION: REQUEST CREATED The contents are 'RequestCreated' and 'https://management.azure.com/subscriptions/422b6c61-95b0-4213-b3be-7282315df71d/resourceGroups/arc-server-test/providers/Microsoft.HybridCompute/machines/tyler-ThinkPad-X1-Carbon/extensions/foo?api-version=2019-12-12'

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
VERBOSE: INFORMATION: REQUEST CREATED Parameter: ''
REQUEST CREATED RequestMessage ''
REQUEST CREATED Response: ''
REQUEST CREATED Value: '0'

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
VERBOSE: INFORMATION: REQUEST CREATED ExtendedData Type: ''
REQUEST CREATED ExtendedData ''

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
DEBUG: RequestCreated: https://management.azure.com/subscriptions/422b6c61-95b0-4213-b3be-7282315df71d/resourceGroups/arc-server-test/providers/Microsoft.HybridCompute/machines/tyler-ThinkPad-X1-Carbon/extensions/foo?api-version=2019-12-12
VERBOSE: INFORMATION: HEADER PARAMETERS ADDED The contents are 'HeaderParametersAdded' and 'https://management.azure.com/subscriptions/422b6c61-95b0-4213-b3be-7282315df71d/resourceGroups/arc-server-test/providers/Microsoft.HybridCompute/machines/tyler-ThinkPad-X1-Carbon/extensions/foo?api-version=2019-12-12'

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
VERBOSE: INFORMATION: HEADER PARAMETERS ADDED Parameter: ''
HEADER PARAMETERS ADDED RequestMessage ''
HEADER PARAMETERS ADDED Response: ''
HEADER PARAMETERS ADDED Value: '0'

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
VERBOSE: INFORMATION: HEADER PARAMETERS ADDED ExtendedData Type: ''
HEADER PARAMETERS ADDED ExtendedData ''

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
DEBUG: HeaderParametersAdded: https://management.azure.com/subscriptions/422b6c61-95b0-4213-b3be-7282315df71d/resourceGroups/arc-server-test/providers/Microsoft.HybridCompute/machines/tyler-ThinkPad-X1-Carbon/extensions/foo?api-version=2019-12-12
VERBOSE: INFORMATION: BODY CONTENT SET The contents are 'BodyContentSet' and 'https://management.azure.com/subscriptions/422b6c61-95b0-4213-b3be-7282315df71d/resourceGroups/arc-server-test/providers/Microsoft.HybridCompute/machines/tyler-ThinkPad-X1-Carbon/extensions/foo?api-version=2019-12-12'

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
VERBOSE: INFORMATION: BODY CONTENT SET Parameter: ''
BODY CONTENT SET RequestMessage ''
BODY CONTENT SET Response: ''
BODY CONTENT SET Value: '0'

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
VERBOSE: INFORMATION: BODY CONTENT SET ExtendedData Type: ''
BODY CONTENT SET ExtendedData ''

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
DEBUG: BodyContentSet: https://management.azure.com/subscriptions/422b6c61-95b0-4213-b3be-7282315df71d/resourceGroups/arc-server-test/providers/Microsoft.HybridCompute/machines/tyler-ThinkPad-X1-Carbon/extensions/foo?api-version=2019-12-12
DEBUG: ============================ HTTP REQUEST ============================

HTTP Method:
PUT

Absolute Uri:
https://management.azure.com/subscriptions/422b6c61-95b0-4213-b3be-7282315df71d/resourceGroups/arc-server-test/providers/Microsoft.HybridCompute/machines/tyler-ThinkPad-X1-Carbon/extensions/foo?api-version=2019-12-12

Headers:

Body:
{
  "location": "eastus",
  "properties": {
    "instanceView": {
      "status": {
        "code": "success",
        "level": "Information",
        "message": "formattedMessage: Enable succeeded: \n[stdout]\nstderr\nstdout\n\n[stderr]\n"
      },
      "name": "foo",
      "type": "CustomScript",
      "typeHandlerVersion": "2.1.3"
    },
    "type": "CustomScript",
    "autoUpgradeMinorVersion": false,
    "publisher": "Microsoft.Azure.Extensions",
    "settings": {
      "commandToExecute": "ls -a"
    },
    "typeHandlerVersion": "2.1.3"
  }
}


DEBUG: BeforeCall: 
DEBUG: [ADAL]: Information: 2020-08-26T05:10:02.1032164Z: 073edd79-58ef-43e5-ab14-8a8cecda349e - LoggerBase.cs: ADAL PCL.CoreCLR with assembly version '3.19.2.6005', file version '3.19.50302.0130' and informational version '2a8bec6c4c76d0c1ef819b55bdc3cda2d2605056' is running...
DEBUG: [ADAL]: Information: 2020-08-26T05:10:02.1033759Z: 073edd79-58ef-43e5-ab14-8a8cecda349e - LoggerBase.cs: ADAL PCL.CoreCLR with assembly version '3.19.2.6005', file version '3.19.50302.0130' and informational version '2a8bec6c4c76d0c1ef819b55bdc3cda2d2605056' is running...
DEBUG: [ADAL]: Information: 2020-08-26T05:10:02.1034778Z: 073edd79-58ef-43e5-ab14-8a8cecda349e - LoggerBase.cs: === Token Acquisition started: 
	CacheType: null
	Authentication Target: User
	, Authority Host: login.microsoftonline.com
DEBUG: [ADAL]: Information: 2020-08-26T05:10:02.1035336Z: 073edd79-58ef-43e5-ab14-8a8cecda349e - LoggerBase.cs: === Token Acquisition started:
	Authority: https://login.microsoftonline.com/72f988bf-86f1-41af-91ab-2d7cd011db47/
	Resource: https://management.core.windows.net/
	ClientId: 1950a258-227b-4e31-a9cf-717495945fc2
	CacheType: null
	Authentication Target: User
	
DEBUG: [ADAL]: Verbose: 2020-08-26T05:10:02.1036685Z: 073edd79-58ef-43e5-ab14-8a8cecda349e - LoggerBase.cs: Loading from cache.
DEBUG: [ADAL]: Verbose: 2020-08-26T05:10:02.1037197Z: 073edd79-58ef-43e5-ab14-8a8cecda349e - LoggerBase.cs: Loading from cache.
DEBUG: [ADAL]: Information: 2020-08-26T05:10:02.1048311Z: 00000000-0000-0000-0000-000000000000 - LoggerBase.cs: Deserialized 3 items to token cache.
DEBUG: [ADAL]: Verbose: 2020-08-26T05:10:02.1049878Z: 073edd79-58ef-43e5-ab14-8a8cecda349e - LoggerBase.cs: Looking up cache for a token...
DEBUG: [ADAL]: Verbose: 2020-08-26T05:10:02.1050598Z: 073edd79-58ef-43e5-ab14-8a8cecda349e - LoggerBase.cs: Looking up cache for a token...
DEBUG: [ADAL]: Information: 2020-08-26T05:10:02.1051668Z: 073edd79-58ef-43e5-ab14-8a8cecda349e - LoggerBase.cs: An item matching the requested resource was found in the cache
DEBUG: [ADAL]: Information: 2020-08-26T05:10:02.1052312Z: 073edd79-58ef-43e5-ab14-8a8cecda349e - LoggerBase.cs: An item matching the requested resource was found in the cache
DEBUG: [ADAL]: Information: 2020-08-26T05:10:02.1053198Z: 073edd79-58ef-43e5-ab14-8a8cecda349e - LoggerBase.cs: 57.401494916666664 minutes left until token in cache expires
DEBUG: [ADAL]: Information: 2020-08-26T05:10:02.1053582Z: 073edd79-58ef-43e5-ab14-8a8cecda349e - LoggerBase.cs: 57.401494916666664 minutes left until token in cache expires
DEBUG: [ADAL]: Information: 2020-08-26T05:10:02.1053898Z: 073edd79-58ef-43e5-ab14-8a8cecda349e - LoggerBase.cs: A matching item (access token or refresh token or both) was found in the cache
DEBUG: [ADAL]: Information: 2020-08-26T05:10:02.1054269Z: 073edd79-58ef-43e5-ab14-8a8cecda349e - LoggerBase.cs: A matching item (access token or refresh token or both) was found in the cache
DEBUG: [ADAL]: Information: 2020-08-26T05:10:02.1055613Z: 073edd79-58ef-43e5-ab14-8a8cecda349e - LoggerBase.cs: === Token Acquisition finished successfully. An access token was returned: Expiration Time: 8/26/2020 6:07:26 AM +00:00
DEBUG: [ADAL]: Information: 2020-08-26T05:10:02.1056130Z: 073edd79-58ef-43e5-ab14-8a8cecda349e - LoggerBase.cs: === Token Acquisition finished successfully. An access token was returned: Expiration Time: 8/26/2020 6:07:26 AM +00:00Access Token Hash: XNvbFj1C+pyfghSu12Vz3fD8JfvYsZTHp4Ff75DRkDA=
	 User id: c256c7b0-e94d-49c4-8585-0c7f3020bd37
DEBUG: ============================ HTTP RESPONSE ============================

Status Code:
Accepted

Headers:
Cache-Control                 : no-cache
Pragma                        : no-cache
Location                      : https://management.azure.com/subscriptions/422b6c61-95b0-4213-b3be-7282315df71d/providers/Microsoft.HybridCompute/locations/eastus/operationresults/372f7387-0f1a-4eb0-a62a-369c779c3cc7?api-version=2019-12-12
Strict-Transport-Security     : max-age=31536000; includeSubDomains
x-ms-request-id               : 7fc67a75-d2b1-4f2d-af48-364100a83ebc
Azure-AsyncOperation          : https://management.azure.com/subscriptions/422b6c61-95b0-4213-b3be-7282315df71d/providers/Microsoft.HybridCompute/locations/eastus/operationstatus/372f7387-0f1a-4eb0-a62a-369c779c3cc7?api-version=2019-12-12
Server                        : Kestrel
x-ms-ratelimit-remaining-subscription-resource-requests: 319
x-ms-correlation-request-id   : 883efb04-8272-4591-93a5-fba69fae8faf
x-ms-routing-request-id       : WESTUS2:20200826T051005Z:883efb04-8272-4591-93a5-fba69fae8faf
X-Content-Type-Options        : nosniff
Date                          : Wed, 26 Aug 2020 05:10:04 GMT

Body:



DEBUG: ResponseCreated: 
VERBOSE: INFORMATION: DELAY BEFORE POLLING The contents are 'DelayBeforePolling' and 'Delaying 30 seconds before polling.'

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
VERBOSE: INFORMATION: DELAY BEFORE POLLING Parameter: ''
DELAY BEFORE POLLING RequestMessage 'Method: PUT, RequestUri: 'https://management.azure.com/subscriptions/422b6c61-95b0-4213-b3be-7282315df71d/resourceGroups/arc-server-test/providers/Microsoft.HybridCompute/machines/tyler-ThinkPad-X1-Carbon/extensions/foo?api-version=2019-12-12', Version: 1.1, Content: System.Net.Http.StringContent, Headers:
{
  x-ms-unique-id: 27
  x-ms-client-request-id: c7c5b66b-b95f-4d23-900b-24a34d04b320
  CommandName: Set-AzConnectedMachineExtension
  FullCommandName: Set-AzConnectedMachineExtension_Update
  ParameterSetName: __AllParameterSets
  User-Agent: AzurePowershell/Az4.0.0-preview
  Authorization: Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsIng1dCI6ImppYk5ia0ZTU2JteFBZck45Q0ZxUms0SzRndyIsImtpZCI6ImppYk5ia0ZTU2JteFBZck45Q0ZxUms0SzRndyJ9.eyJhdWQiOiJodHRwczovL21hbmFnZW1lbnQuY29yZS53aW5kb3dzLm5ldC8iLCJpc3MiOiJodHRwczovL3N0cy53aW5kb3dzLm5ldC83MmY5ODhiZi04NmYxLTQxYWYtOTFhYi0yZDdjZDAxMWRiNDcvIiwiaWF0IjoxNTk4NDE4MTQ4LCJuYmYiOjE1OTg0MTgxNDgsImV4cCI6MTU5ODQyMjA0OCwiX2NsYWltX25hbWVzIjp7Imdyb3VwcyI6InNyYzEifSwiX2NsYWltX3NvdXJjZXMiOnsic3JjMSI6eyJlbmRwb2ludCI6Imh0dHBzOi8vZ3JhcGgud2luZG93cy5uZXQvNzJmOTg4YmYtODZmMS00MWFmLTkxYWItMmQ3Y2QwMTFkYjQ3L3VzZXJzL2MyNTZjN2IwLWU5NGQtNDljNC04NTg1LTBjN2YzMDIwYmQzNy9nZXRNZW1iZXJPYmplY3RzIn19LCJhY3IiOiIxIiwiYWlvIjoiQVZRQXEvOFFBQUFBcnNiRnYvMU5kdXVpcjJuNy9nYUkzYkw4SkVNY3E1YnVlOEtZMGE3bk5VK3YwOFpIT1YzTG5EZGxqWkxrZUZRbk52OEJ5NUlMZkpIY3ErN29vaEFLQmx1bUV4Z1NNdis2OXRPVTRWa3pOcFE9IiwiYW1yIjpbInB3ZCIsIm1mYSJdLCJhcHBpZCI6IjE5NTBhMjU4LTIyN2ItNGUzMS1hOWNmLTcxNzQ5NTk0NWZjMiIsImFwcGlkYWNyIjoiMCIsImZhbWlseV9uYW1lIjoiTGVvbmhhcmR0IChQT1dFUlNIRUxMKSIsImdpdmVuX25hbWUiOiJUeWxlciIsImlwYWRkciI6Ijc1LjE3Mi4xODMuMTc2IiwibmFtZSI6IlR5bGVyIExlb25oYXJkdCAoUE9XRVJTSEVMTCkiLCJvaWQiOiJjMjU2YzdiMC1lOTRkLTQ5YzQtODU4NS0wYzdmMzAyMGJkMzciLCJvbnByZW1fc2lkIjoiUy0xLTUtMjEtMjEyNzUyMTE4NC0xNjA0MDEyOTIwLTE4ODc5Mjc1MjctMjIyNjkzMzgiLCJwdWlkIjoiMTAwMzdGRkU5OThDRkZGNyIsInJoIjoiMC5BUm9BdjRqNWN2R0dyMEdScXkxODBCSGJSMWlpVUJsN0lqRk9xYzl4ZEpXVVg4SWFBTk0uIiwic2NwIjoidXNlcl9pbXBlcnNvbmF0aW9uIiwic3ViIjoiMXJrT05YeDAtUDVJQnlFc2kyV1RROVZXQUtmNW9IQjRVRnNkV040aFlBSSIsInRpZCI6IjcyZjk4OGJmLTg2ZjEtNDFhZi05MWFiLTJkN2NkMDExZGI0NyIsInVuaXF1ZV9uYW1lIjoidHlsZW9uaGFAbWljcm9zb2Z0LmNvbSIsInVwbiI6InR5bGVvbmhhQG1pY3Jvc29mdC5jb20iLCJ1dGkiOiJZYzF3TWtsUUYwMkRiU1k5RXR3TEFBIiwidmVyIjoiMS4wIiwieG1zX3RjZHQiOjEyODkyNDE1NDd9.VbmeQcgINWvA1hIQ2IOZh1ND1QpY0qwo5UQm21Y2zXTJ6ah1PEJN4iSWOxk8KWsc7lpNB3BBEWtgoT4YDhE79kmFZl-miiw2CRTstzrYLSUVD4vG_yAhjJ6b7q3bgv_fnvioZ6TekJ7J9DNQWTLC_IVaD57-vnN8JDbG-GAD356Zl_ywBKGhZoJkoJPK-O_zb73SpP6kUw8uxa49UGqI3qt7VVAVYRIkSnPucLZu5VjwEBSGxzVpuUANh7Ag7Pz8k1ZXEupjTcsF2kcba3vVumtHm90PmpXoJlkJDQV8os7MpDaRJOcbc1p5FlzpiReL70HXxumZB2H5HmbCbucX1A
  Content-Type: application/json
  Content-Length: 555
}'
DELAY BEFORE POLLING Response: 'StatusCode: 202, ReasonPhrase: 'Accepted', Version: 1.1, Content: System.Net.Http.HttpConnectionResponseContent, Headers:
{
  Cache-Control: no-cache
  Pragma: no-cache
  Location: https://management.azure.com/subscriptions/422b6c61-95b0-4213-b3be-7282315df71d/providers/Microsoft.HybridCompute/locations/eastus/operationresults/372f7387-0f1a-4eb0-a62a-369c779c3cc7?api-version=2019-12-12
  Strict-Transport-Security: max-age=31536000; includeSubDomains
  x-ms-request-id: 7fc67a75-d2b1-4f2d-af48-364100a83ebc
  Azure-AsyncOperation: https://management.azure.com/subscriptions/422b6c61-95b0-4213-b3be-7282315df71d/providers/Microsoft.HybridCompute/locations/eastus/operationstatus/372f7387-0f1a-4eb0-a62a-369c779c3cc7?api-version=2019-12-12
  Server: Kestrel
  x-ms-ratelimit-remaining-subscription-resource-requests: 319
  x-ms-correlation-request-id: 883efb04-8272-4591-93a5-fba69fae8faf
  x-ms-routing-request-id: WESTUS2:20200826T051005Z:883efb04-8272-4591-93a5-fba69fae8faf
  X-Content-Type-Options: nosniff
  Date: Wed, 26 Aug 2020 05:10:04 GMT
  Content-Type: application/json
  Expires: -1
  Content-Length: 0
}'
DELAY BEFORE POLLING Value: '0'

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
VERBOSE: INFORMATION: DELAY BEFORE POLLING ExtendedData Type: ''
DELAY BEFORE POLLING ExtendedData ''

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
DEBUG: DelayBeforePolling: Delaying 30 seconds before polling.
VERBOSE: INFORMATION: POLLING The contents are 'Polling' and 'Polling https://management.azure.com/subscriptions/422b6c61-95b0-4213-b3be-7282315df71d/providers/Microsoft.HybridCompute/locations/eastus/operationstatus/372f7387-0f1a-4eb0-a62a-369c779c3cc7?api-version=2019-12-12.'

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
VERBOSE: INFORMATION: POLLING Parameter: ''
POLLING RequestMessage 'Method: PUT, RequestUri: 'https://management.azure.com/subscriptions/422b6c61-95b0-4213-b3be-7282315df71d/resourceGroups/arc-server-test/providers/Microsoft.HybridCompute/machines/tyler-ThinkPad-X1-Carbon/extensions/foo?api-version=2019-12-12', Version: 1.1, Content: System.Net.Http.StringContent, Headers:
{
  x-ms-unique-id: 27
  x-ms-client-request-id: c7c5b66b-b95f-4d23-900b-24a34d04b320
  CommandName: Set-AzConnectedMachineExtension
  FullCommandName: Set-AzConnectedMachineExtension_Update
  ParameterSetName: __AllParameterSets
  User-Agent: AzurePowershell/Az4.0.0-preview
  Authorization: Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsIng1dCI6ImppYk5ia0ZTU2JteFBZck45Q0ZxUms0SzRndyIsImtpZCI6ImppYk5ia0ZTU2JteFBZck45Q0ZxUms0SzRndyJ9.eyJhdWQiOiJodHRwczovL21hbmFnZW1lbnQuY29yZS53aW5kb3dzLm5ldC8iLCJpc3MiOiJodHRwczovL3N0cy53aW5kb3dzLm5ldC83MmY5ODhiZi04NmYxLTQxYWYtOTFhYi0yZDdjZDAxMWRiNDcvIiwiaWF0IjoxNTk4NDE4MTQ4LCJuYmYiOjE1OTg0MTgxNDgsImV4cCI6MTU5ODQyMjA0OCwiX2NsYWltX25hbWVzIjp7Imdyb3VwcyI6InNyYzEifSwiX2NsYWltX3NvdXJjZXMiOnsic3JjMSI6eyJlbmRwb2ludCI6Imh0dHBzOi8vZ3JhcGgud2luZG93cy5uZXQvNzJmOTg4YmYtODZmMS00MWFmLTkxYWItMmQ3Y2QwMTFkYjQ3L3VzZXJzL2MyNTZjN2IwLWU5NGQtNDljNC04NTg1LTBjN2YzMDIwYmQzNy9nZXRNZW1iZXJPYmplY3RzIn19LCJhY3IiOiIxIiwiYWlvIjoiQVZRQXEvOFFBQUFBcnNiRnYvMU5kdXVpcjJuNy9nYUkzYkw4SkVNY3E1YnVlOEtZMGE3bk5VK3YwOFpIT1YzTG5EZGxqWkxrZUZRbk52OEJ5NUlMZkpIY3ErN29vaEFLQmx1bUV4Z1NNdis2OXRPVTRWa3pOcFE9IiwiYW1yIjpbInB3ZCIsIm1mYSJdLCJhcHBpZCI6IjE5NTBhMjU4LTIyN2ItNGUzMS1hOWNmLTcxNzQ5NTk0NWZjMiIsImFwcGlkYWNyIjoiMCIsImZhbWlseV9uYW1lIjoiTGVvbmhhcmR0IChQT1dFUlNIRUxMKSIsImdpdmVuX25hbWUiOiJUeWxlciIsImlwYWRkciI6Ijc1LjE3Mi4xODMuMTc2IiwibmFtZSI6IlR5bGVyIExlb25oYXJkdCAoUE9XRVJTSEVMTCkiLCJvaWQiOiJjMjU2YzdiMC1lOTRkLTQ5YzQtODU4NS0wYzdmMzAyMGJkMzciLCJvbnByZW1fc2lkIjoiUy0xLTUtMjEtMjEyNzUyMTE4NC0xNjA0MDEyOTIwLTE4ODc5Mjc1MjctMjIyNjkzMzgiLCJwdWlkIjoiMTAwMzdGRkU5OThDRkZGNyIsInJoIjoiMC5BUm9BdjRqNWN2R0dyMEdScXkxODBCSGJSMWlpVUJsN0lqRk9xYzl4ZEpXVVg4SWFBTk0uIiwic2NwIjoidXNlcl9pbXBlcnNvbmF0aW9uIiwic3ViIjoiMXJrT05YeDAtUDVJQnlFc2kyV1RROVZXQUtmNW9IQjRVRnNkV040aFlBSSIsInRpZCI6IjcyZjk4OGJmLTg2ZjEtNDFhZi05MWFiLTJkN2NkMDExZGI0NyIsInVuaXF1ZV9uYW1lIjoidHlsZW9uaGFAbWljcm9zb2Z0LmNvbSIsInVwbiI6InR5bGVvbmhhQG1pY3Jvc29mdC5jb20iLCJ1dGkiOiJZYzF3TWtsUUYwMkRiU1k5RXR3TEFBIiwidmVyIjoiMS4wIiwieG1zX3RjZHQiOjEyODkyNDE1NDd9.VbmeQcgINWvA1hIQ2IOZh1ND1QpY0qwo5UQm21Y2zXTJ6ah1PEJN4iSWOxk8KWsc7lpNB3BBEWtgoT4YDhE79kmFZl-miiw2CRTstzrYLSUVD4vG_yAhjJ6b7q3bgv_fnvioZ6TekJ7J9DNQWTLC_IVaD57-vnN8JDbG-GAD356Zl_ywBKGhZoJkoJPK-O_zb73SpP6kUw8uxa49UGqI3qt7VVAVYRIkSnPucLZu5VjwEBSGxzVpuUANh7Ag7Pz8k1ZXEupjTcsF2kcba3vVumtHm90PmpXoJlkJDQV8os7MpDaRJOcbc1p5FlzpiReL70HXxumZB2H5HmbCbucX1A
  Content-Type: application/json
  Content-Length: 555
}'
POLLING Response: 'StatusCode: 202, ReasonPhrase: 'Accepted', Version: 1.1, Content: System.Net.Http.HttpConnectionResponseContent, Headers:
{
  Cache-Control: no-cache
  Pragma: no-cache
  Location: https://management.azure.com/subscriptions/422b6c61-95b0-4213-b3be-7282315df71d/providers/Microsoft.HybridCompute/locations/eastus/operationresults/372f7387-0f1a-4eb0-a62a-369c779c3cc7?api-version=2019-12-12
  Strict-Transport-Security: max-age=31536000; includeSubDomains
  x-ms-request-id: 7fc67a75-d2b1-4f2d-af48-364100a83ebc
  Azure-AsyncOperation: https://management.azure.com/subscriptions/422b6c61-95b0-4213-b3be-7282315df71d/providers/Microsoft.HybridCompute/locations/eastus/operationstatus/372f7387-0f1a-4eb0-a62a-369c779c3cc7?api-version=2019-12-12
  Server: Kestrel
  x-ms-ratelimit-remaining-subscription-resource-requests: 319
  x-ms-correlation-request-id: 883efb04-8272-4591-93a5-fba69fae8faf
  x-ms-routing-request-id: WESTUS2:20200826T051005Z:883efb04-8272-4591-93a5-fba69fae8faf
  X-Content-Type-Options: nosniff
  Date: Wed, 26 Aug 2020 05:10:04 GMT
  Content-Type: application/json
  Expires: -1
  Content-Length: 0
}'
POLLING Value: '0'

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
VERBOSE: INFORMATION: POLLING ExtendedData Type: ''
POLLING ExtendedData ''

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
DEBUG: Polling: Polling https://management.azure.com/subscriptions/422b6c61-95b0-4213-b3be-7282315df71d/providers/Microsoft.HybridCompute/locations/eastus/operationstatus/372f7387-0f1a-4eb0-a62a-369c779c3cc7?api-version=2019-12-12.
DEBUG: [ADAL]: Information: 2020-08-26T05:17:03.3011534Z: 461e0005-3284-4ea9-89dd-b0eb126c45d2 - LoggerBase.cs: ADAL PCL.CoreCLR with assembly version '3.19.2.6005', file version '3.19.50302.0130' and informational version '2a8bec6c4c76d0c1ef819b55bdc3cda2d2605056' is running...
DEBUG: [ADAL]: Information: 2020-08-26T05:17:03.3013201Z: 461e0005-3284-4ea9-89dd-b0eb126c45d2 - LoggerBase.cs: ADAL PCL.CoreCLR with assembly version '3.19.2.6005', file version '3.19.50302.0130' and informational version '2a8bec6c4c76d0c1ef819b55bdc3cda2d2605056' is running...
DEBUG: [ADAL]: Information: 2020-08-26T05:17:03.3014510Z: 461e0005-3284-4ea9-89dd-b0eb126c45d2 - LoggerBase.cs: === Token Acquisition started: 
	CacheType: null
	Authentication Target: User
	, Authority Host: login.microsoftonline.com
DEBUG: [ADAL]: Information: 2020-08-26T05:17:03.3015502Z: 461e0005-3284-4ea9-89dd-b0eb126c45d2 - LoggerBase.cs: === Token Acquisition started:
	Authority: https://login.microsoftonline.com/72f988bf-86f1-41af-91ab-2d7cd011db47/
	Resource: https://management.core.windows.net/
	ClientId: 1950a258-227b-4e31-a9cf-717495945fc2
	CacheType: null
	Authentication Target: User
	
DEBUG: [ADAL]: Verbose: 2020-08-26T05:17:03.3017349Z: 461e0005-3284-4ea9-89dd-b0eb126c45d2 - LoggerBase.cs: Loading from cache.
DEBUG: [ADAL]: Verbose: 2020-08-26T05:17:03.3018153Z: 461e0005-3284-4ea9-89dd-b0eb126c45d2 - LoggerBase.cs: Loading from cache.
DEBUG: [ADAL]: Information: 2020-08-26T05:17:03.3031761Z: 00000000-0000-0000-0000-000000000000 - LoggerBase.cs: Deserialized 3 items to token cache.
DEBUG: [ADAL]: Verbose: 2020-08-26T05:17:03.3032263Z: 461e0005-3284-4ea9-89dd-b0eb126c45d2 - LoggerBase.cs: Looking up cache for a token...
DEBUG: [ADAL]: Verbose: 2020-08-26T05:17:03.3032465Z: 461e0005-3284-4ea9-89dd-b0eb126c45d2 - LoggerBase.cs: Looking up cache for a token...
DEBUG: [ADAL]: Information: 2020-08-26T05:17:03.3032812Z: 461e0005-3284-4ea9-89dd-b0eb126c45d2 - LoggerBase.cs: An item matching the requested resource was found in the cache
DEBUG: [ADAL]: Information: 2020-08-26T05:17:03.3032992Z: 461e0005-3284-4ea9-89dd-b0eb126c45d2 - LoggerBase.cs: An item matching the requested resource was found in the cache
DEBUG: [ADAL]: Information: 2020-08-26T05:17:03.3033287Z: 461e0005-3284-4ea9-89dd-b0eb126c45d2 - LoggerBase.cs: 50.381527965 minutes left until token in cache expires
DEBUG: [ADAL]: Information: 2020-08-26T05:17:03.3033656Z: 461e0005-3284-4ea9-89dd-b0eb126c45d2 - LoggerBase.cs: 50.381527965 minutes left until token in cache expires
DEBUG: [ADAL]: Information: 2020-08-26T05:17:03.3034187Z: 461e0005-3284-4ea9-89dd-b0eb126c45d2 - LoggerBase.cs: A matching item (access token or refresh token or both) was found in the cache
DEBUG: [ADAL]: Information: 2020-08-26T05:17:03.3034769Z: 461e0005-3284-4ea9-89dd-b0eb126c45d2 - LoggerBase.cs: A matching item (access token or refresh token or both) was found in the cache
DEBUG: [ADAL]: Information: 2020-08-26T05:17:03.3036716Z: 461e0005-3284-4ea9-89dd-b0eb126c45d2 - LoggerBase.cs: === Token Acquisition finished successfully. An access token was returned: Expiration Time: 8/26/2020 6:07:26 AM +00:00
DEBUG: [ADAL]: Information: 2020-08-26T05:17:03.3037473Z: 461e0005-3284-4ea9-89dd-b0eb126c45d2 - LoggerBase.cs: === Token Acquisition finished successfully. An access token was returned: Expiration Time: 8/26/2020 6:07:26 AM +00:00Access Token Hash: XNvbFj1C+pyfghSu12Vz3fD8JfvYsZTHp4Ff75DRkDA=
	 User id: c256c7b0-e94d-49c4-8585-0c7f3020bd37
VERBOSE: INFORMATION: POLLING The contents are 'Polling' and 'Polled https://management.azure.com/subscriptions/422b6c61-95b0-4213-b3be-7282315df71d/providers/Microsoft.HybridCompute/locations/eastus/operationstatus/372f7387-0f1a-4eb0-a62a-369c779c3cc7?api-version=2019-12-12 provisioning state  Failed.'

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
VERBOSE: INFORMATION: POLLING Parameter: ''
POLLING RequestMessage 'Method: GET, RequestUri: 'https://management.azure.com/subscriptions/422b6c61-95b0-4213-b3be-7282315df71d/providers/Microsoft.HybridCompute/locations/eastus/operationstatus/372f7387-0f1a-4eb0-a62a-369c779c3cc7?api-version=2019-12-12', Version: 1.1, Content: <null>, Headers:
{
  x-ms-unique-id: 27
  x-ms-unique-id: 28
  x-ms-client-request-id: c7c5b66b-b95f-4d23-900b-24a34d04b320
  x-ms-client-request-id: c7c5b66b-b95f-4d23-900b-24a34d04b320
  CommandName: Set-AzConnectedMachineExtension
  CommandName: Set-AzConnectedMachineExtension
  FullCommandName: Set-AzConnectedMachineExtension_Update
  FullCommandName: Set-AzConnectedMachineExtension_Update
  ParameterSetName: __AllParameterSets
  ParameterSetName: __AllParameterSets
  User-Agent: AzurePowershell/Az4.0.0-preview
  User-Agent: AzurePowershell/Az4.0.0-preview
  Authorization: Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsIng1dCI6ImppYk5ia0ZTU2JteFBZck45Q0ZxUms0SzRndyIsImtpZCI6ImppYk5ia0ZTU2JteFBZck45Q0ZxUms0SzRndyJ9.eyJhdWQiOiJodHRwczovL21hbmFnZW1lbnQuY29yZS53aW5kb3dzLm5ldC8iLCJpc3MiOiJodHRwczovL3N0cy53aW5kb3dzLm5ldC83MmY5ODhiZi04NmYxLTQxYWYtOTFhYi0yZDdjZDAxMWRiNDcvIiwiaWF0IjoxNTk4NDE4MTQ4LCJuYmYiOjE1OTg0MTgxNDgsImV4cCI6MTU5ODQyMjA0OCwiX2NsYWltX25hbWVzIjp7Imdyb3VwcyI6InNyYzEifSwiX2NsYWltX3NvdXJjZXMiOnsic3JjMSI6eyJlbmRwb2ludCI6Imh0dHBzOi8vZ3JhcGgud2luZG93cy5uZXQvNzJmOTg4YmYtODZmMS00MWFmLTkxYWItMmQ3Y2QwMTFkYjQ3L3VzZXJzL2MyNTZjN2IwLWU5NGQtNDljNC04NTg1LTBjN2YzMDIwYmQzNy9nZXRNZW1iZXJPYmplY3RzIn19LCJhY3IiOiIxIiwiYWlvIjoiQVZRQXEvOFFBQUFBcnNiRnYvMU5kdXVpcjJuNy9nYUkzYkw4SkVNY3E1YnVlOEtZMGE3bk5VK3YwOFpIT1YzTG5EZGxqWkxrZUZRbk52OEJ5NUlMZkpIY3ErN29vaEFLQmx1bUV4Z1NNdis2OXRPVTRWa3pOcFE9IiwiYW1yIjpbInB3ZCIsIm1mYSJdLCJhcHBpZCI6IjE5NTBhMjU4LTIyN2ItNGUzMS1hOWNmLTcxNzQ5NTk0NWZjMiIsImFwcGlkYWNyIjoiMCIsImZhbWlseV9uYW1lIjoiTGVvbmhhcmR0IChQT1dFUlNIRUxMKSIsImdpdmVuX25hbWUiOiJUeWxlciIsImlwYWRkciI6Ijc1LjE3Mi4xODMuMTc2IiwibmFtZSI6IlR5bGVyIExlb25oYXJkdCAoUE9XRVJTSEVMTCkiLCJvaWQiOiJjMjU2YzdiMC1lOTRkLTQ5YzQtODU4NS0wYzdmMzAyMGJkMzciLCJvbnByZW1fc2lkIjoiUy0xLTUtMjEtMjEyNzUyMTE4NC0xNjA0MDEyOTIwLTE4ODc5Mjc1MjctMjIyNjkzMzgiLCJwdWlkIjoiMTAwMzdGRkU5OThDRkZGNyIsInJoIjoiMC5BUm9BdjRqNWN2R0dyMEdScXkxODBCSGJSMWlpVUJsN0lqRk9xYzl4ZEpXVVg4SWFBTk0uIiwic2NwIjoidXNlcl9pbXBlcnNvbmF0aW9uIiwic3ViIjoiMXJrT05YeDAtUDVJQnlFc2kyV1RROVZXQUtmNW9IQjRVRnNkV040aFlBSSIsInRpZCI6IjcyZjk4OGJmLTg2ZjEtNDFhZi05MWFiLTJkN2NkMDExZGI0NyIsInVuaXF1ZV9uYW1lIjoidHlsZW9uaGFAbWljcm9zb2Z0LmNvbSIsInVwbiI6InR5bGVvbmhhQG1pY3Jvc29mdC5jb20iLCJ1dGkiOiJZYzF3TWtsUUYwMkRiU1k5RXR3TEFBIiwidmVyIjoiMS4wIiwieG1zX3RjZHQiOjEyODkyNDE1NDd9.VbmeQcgINWvA1hIQ2IOZh1ND1QpY0qwo5UQm21Y2zXTJ6ah1PEJN4iSWOxk8KWsc7lpNB3BBEWtgoT4YDhE79kmFZl-miiw2CRTstzrYLSUVD4vG_yAhjJ6b7q3bgv_fnvioZ6TekJ7J9DNQWTLC_IVaD57-vnN8JDbG-GAD356Zl_ywBKGhZoJkoJPK-O_zb73SpP6kUw8uxa49UGqI3qt7VVAVYRIkSnPucLZu5VjwEBSGxzVpuUANh7Ag7Pz8k1ZXEupjTcsF2kcba3vVumtHm90PmpXoJlkJDQV8os7MpDaRJOcbc1p5FlzpiReL70HXxumZB2H5HmbCbucX1A
}'
POLLING Response: 'StatusCode: 200, ReasonPhrase: 'OK', Version: 1.1, Content: System.Net.Http.HttpConnectionResponseContent, Headers:
{
  Cache-Control: no-cache
  Pragma: no-cache
  x-ms-ratelimit-remaining-subscription-reads: 11999
  Strict-Transport-Security: max-age=31536000; includeSubDomains
  x-ms-request-id: 1584595a-d876-40ef-bdad-b8726c1bd99b
  Server: Kestrel
  x-ms-correlation-request-id: a253dafa-d797-4622-9d22-4a895e4bde37
  x-ms-routing-request-id: WESTUS2:20200826T051705Z:a253dafa-d797-4622-9d22-4a895e4bde37
  X-Content-Type-Options: nosniff
  Date: Wed, 26 Aug 2020 05:17:04 GMT
  Content-Length: 159
  Content-Type: application/json; charset=utf-8
  Expires: -1
}'
POLLING Value: '0'

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
VERBOSE: INFORMATION: POLLING ExtendedData Type: ''
POLLING ExtendedData ''

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
DEBUG: Polling: Polled https://management.azure.com/subscriptions/422b6c61-95b0-4213-b3be-7282315df71d/providers/Microsoft.HybridCompute/locations/eastus/operationstatus/372f7387-0f1a-4eb0-a62a-369c779c3cc7?api-version=2019-12-12 provisioning state  Failed.
DEBUG: [ADAL]: Information: 2020-08-26T05:17:19.3284484Z: f9422e70-c545-498c-a93c-a611d60f7493 - LoggerBase.cs: ADAL PCL.CoreCLR with assembly version '3.19.2.6005', file version '3.19.50302.0130' and informational version '2a8bec6c4c76d0c1ef819b55bdc3cda2d2605056' is running...
DEBUG: [ADAL]: Information: 2020-08-26T05:17:19.3286810Z: f9422e70-c545-498c-a93c-a611d60f7493 - LoggerBase.cs: ADAL PCL.CoreCLR with assembly version '3.19.2.6005', file version '3.19.50302.0130' and informational version '2a8bec6c4c76d0c1ef819b55bdc3cda2d2605056' is running...
DEBUG: [ADAL]: Information: 2020-08-26T05:17:19.3288494Z: f9422e70-c545-498c-a93c-a611d60f7493 - LoggerBase.cs: === Token Acquisition started: 
	CacheType: null
	Authentication Target: User
	, Authority Host: login.microsoftonline.com
DEBUG: [ADAL]: Information: 2020-08-26T05:17:19.3289485Z: f9422e70-c545-498c-a93c-a611d60f7493 - LoggerBase.cs: === Token Acquisition started:
	Authority: https://login.microsoftonline.com/72f988bf-86f1-41af-91ab-2d7cd011db47/
	Resource: https://management.core.windows.net/
	ClientId: 1950a258-227b-4e31-a9cf-717495945fc2
	CacheType: null
	Authentication Target: User
	
DEBUG: [ADAL]: Verbose: 2020-08-26T05:17:19.3295885Z: f9422e70-c545-498c-a93c-a611d60f7493 - LoggerBase.cs: Loading from cache.
DEBUG: [ADAL]: Verbose: 2020-08-26T05:17:19.3296732Z: f9422e70-c545-498c-a93c-a611d60f7493 - LoggerBase.cs: Loading from cache.
DEBUG: [ADAL]: Information: 2020-08-26T05:17:19.3311474Z: 00000000-0000-0000-0000-000000000000 - LoggerBase.cs: Deserialized 3 items to token cache.
DEBUG: [ADAL]: Verbose: 2020-08-26T05:17:19.3312096Z: f9422e70-c545-498c-a93c-a611d60f7493 - LoggerBase.cs: Looking up cache for a token...
DEBUG: [ADAL]: Verbose: 2020-08-26T05:17:19.3312321Z: f9422e70-c545-498c-a93c-a611d60f7493 - LoggerBase.cs: Looking up cache for a token...
DEBUG: [ADAL]: Information: 2020-08-26T05:17:19.3312659Z: f9422e70-c545-498c-a93c-a611d60f7493 - LoggerBase.cs: An item matching the requested resource was found in the cache
DEBUG: [ADAL]: Information: 2020-08-26T05:17:19.3312858Z: f9422e70-c545-498c-a93c-a611d60f7493 - LoggerBase.cs: An item matching the requested resource was found in the cache
DEBUG: [ADAL]: Information: 2020-08-26T05:17:19.3313274Z: f9422e70-c545-498c-a93c-a611d60f7493 - LoggerBase.cs: 50.114394665 minutes left until token in cache expires
DEBUG: [ADAL]: Information: 2020-08-26T05:17:19.3313471Z: f9422e70-c545-498c-a93c-a611d60f7493 - LoggerBase.cs: 50.114394665 minutes left until token in cache expires
DEBUG: [ADAL]: Information: 2020-08-26T05:17:19.3313639Z: f9422e70-c545-498c-a93c-a611d60f7493 - LoggerBase.cs: A matching item (access token or refresh token or both) was found in the cache
DEBUG: [ADAL]: Information: 2020-08-26T05:17:19.3313824Z: f9422e70-c545-498c-a93c-a611d60f7493 - LoggerBase.cs: A matching item (access token or refresh token or both) was found in the cache
DEBUG: [ADAL]: Information: 2020-08-26T05:17:19.3314566Z: f9422e70-c545-498c-a93c-a611d60f7493 - LoggerBase.cs: === Token Acquisition finished successfully. An access token was returned: Expiration Time: 8/26/2020 6:07:26 AM +00:00
DEBUG: [ADAL]: Information: 2020-08-26T05:17:19.3314810Z: f9422e70-c545-498c-a93c-a611d60f7493 - LoggerBase.cs: === Token Acquisition finished successfully. An access token was returned: Expiration Time: 8/26/2020 6:07:26 AM +00:00Access Token Hash: XNvbFj1C+pyfghSu12Vz3fD8JfvYsZTHp4Ff75DRkDA=
	 User id: c256c7b0-e94d-49c4-8585-0c7f3020bd37
VERBOSE: INFORMATION: BEFORE RESPONSE DISPATCH The contents are 'BeforeResponseDispatch' and ''

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
VERBOSE: INFORMATION: BEFORE RESPONSE DISPATCH Parameter: ''
BEFORE RESPONSE DISPATCH RequestMessage 'Method: GET, RequestUri: 'https://management.azure.com/subscriptions/422b6c61-95b0-4213-b3be-7282315df71d/resourceGroups/arc-server-test/providers/Microsoft.HybridCompute/machines/tyler-ThinkPad-X1-Carbon/extensions/foo?api-version=2019-12-12', Version: 1.1, Content: <null>, Headers:
{
  x-ms-unique-id: 27
  x-ms-unique-id: 28
  x-ms-unique-id: 29
  x-ms-client-request-id: c7c5b66b-b95f-4d23-900b-24a34d04b320
  x-ms-client-request-id: c7c5b66b-b95f-4d23-900b-24a34d04b320
  x-ms-client-request-id: c7c5b66b-b95f-4d23-900b-24a34d04b320
  CommandName: Set-AzConnectedMachineExtension
  CommandName: Set-AzConnectedMachineExtension
  CommandName: Set-AzConnectedMachineExtension
  FullCommandName: Set-AzConnectedMachineExtension_Update
  FullCommandName: Set-AzConnectedMachineExtension_Update
  FullCommandName: Set-AzConnectedMachineExtension_Update
  ParameterSetName: __AllParameterSets
  ParameterSetName: __AllParameterSets
  ParameterSetName: __AllParameterSets
  User-Agent: AzurePowershell/Az4.0.0-preview
  User-Agent: AzurePowershell/Az4.0.0-preview
  User-Agent: AzurePowershell/Az4.0.0-preview
  Authorization: Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsIng1dCI6ImppYk5ia0ZTU2JteFBZck45Q0ZxUms0SzRndyIsImtpZCI6ImppYk5ia0ZTU2JteFBZck45Q0ZxUms0SzRndyJ9.eyJhdWQiOiJodHRwczovL21hbmFnZW1lbnQuY29yZS53aW5kb3dzLm5ldC8iLCJpc3MiOiJodHRwczovL3N0cy53aW5kb3dzLm5ldC83MmY5ODhiZi04NmYxLTQxYWYtOTFhYi0yZDdjZDAxMWRiNDcvIiwiaWF0IjoxNTk4NDE4MTQ4LCJuYmYiOjE1OTg0MTgxNDgsImV4cCI6MTU5ODQyMjA0OCwiX2NsYWltX25hbWVzIjp7Imdyb3VwcyI6InNyYzEifSwiX2NsYWltX3NvdXJjZXMiOnsic3JjMSI6eyJlbmRwb2ludCI6Imh0dHBzOi8vZ3JhcGgud2luZG93cy5uZXQvNzJmOTg4YmYtODZmMS00MWFmLTkxYWItMmQ3Y2QwMTFkYjQ3L3VzZXJzL2MyNTZjN2IwLWU5NGQtNDljNC04NTg1LTBjN2YzMDIwYmQzNy9nZXRNZW1iZXJPYmplY3RzIn19LCJhY3IiOiIxIiwiYWlvIjoiQVZRQXEvOFFBQUFBcnNiRnYvMU5kdXVpcjJuNy9nYUkzYkw4SkVNY3E1YnVlOEtZMGE3bk5VK3YwOFpIT1YzTG5EZGxqWkxrZUZRbk52OEJ5NUlMZkpIY3ErN29vaEFLQmx1bUV4Z1NNdis2OXRPVTRWa3pOcFE9IiwiYW1yIjpbInB3ZCIsIm1mYSJdLCJhcHBpZCI6IjE5NTBhMjU4LTIyN2ItNGUzMS1hOWNmLTcxNzQ5NTk0NWZjMiIsImFwcGlkYWNyIjoiMCIsImZhbWlseV9uYW1lIjoiTGVvbmhhcmR0IChQT1dFUlNIRUxMKSIsImdpdmVuX25hbWUiOiJUeWxlciIsImlwYWRkciI6Ijc1LjE3Mi4xODMuMTc2IiwibmFtZSI6IlR5bGVyIExlb25oYXJkdCAoUE9XRVJTSEVMTCkiLCJvaWQiOiJjMjU2YzdiMC1lOTRkLTQ5YzQtODU4NS0wYzdmMzAyMGJkMzciLCJvbnByZW1fc2lkIjoiUy0xLTUtMjEtMjEyNzUyMTE4NC0xNjA0MDEyOTIwLTE4ODc5Mjc1MjctMjIyNjkzMzgiLCJwdWlkIjoiMTAwMzdGRkU5OThDRkZGNyIsInJoIjoiMC5BUm9BdjRqNWN2R0dyMEdScXkxODBCSGJSMWlpVUJsN0lqRk9xYzl4ZEpXVVg4SWFBTk0uIiwic2NwIjoidXNlcl9pbXBlcnNvbmF0aW9uIiwic3ViIjoiMXJrT05YeDAtUDVJQnlFc2kyV1RROVZXQUtmNW9IQjRVRnNkV040aFlBSSIsInRpZCI6IjcyZjk4OGJmLTg2ZjEtNDFhZi05MWFiLTJkN2NkMDExZGI0NyIsInVuaXF1ZV9uYW1lIjoidHlsZW9uaGFAbWljcm9zb2Z0LmNvbSIsInVwbiI6InR5bGVvbmhhQG1pY3Jvc29mdC5jb20iLCJ1dGkiOiJZYzF3TWtsUUYwMkRiU1k5RXR3TEFBIiwidmVyIjoiMS4wIiwieG1zX3RjZHQiOjEyODkyNDE1NDd9.VbmeQcgINWvA1hIQ2IOZh1ND1QpY0qwo5UQm21Y2zXTJ6ah1PEJN4iSWOxk8KWsc7lpNB3BBEWtgoT4YDhE79kmFZl-miiw2CRTstzrYLSUVD4vG_yAhjJ6b7q3bgv_fnvioZ6TekJ7J9DNQWTLC_IVaD57-vnN8JDbG-GAD356Zl_ywBKGhZoJkoJPK-O_zb73SpP6kUw8uxa49UGqI3qt7VVAVYRIkSnPucLZu5VjwEBSGxzVpuUANh7Ag7Pz8k1ZXEupjTcsF2kcba3vVumtHm90PmpXoJlkJDQV8os7MpDaRJOcbc1p5FlzpiReL70HXxumZB2H5HmbCbucX1A
}'
BEFORE RESPONSE DISPATCH Response: 'StatusCode: 200, ReasonPhrase: 'OK', Version: 1.1, Content: System.Net.Http.HttpConnectionResponseContent, Headers:
{
  Cache-Control: no-cache
  Pragma: no-cache
  x-ms-ratelimit-remaining-subscription-reads: 11998
  Strict-Transport-Security: max-age=31536000; includeSubDomains
  x-ms-request-id: f68bd6e3-a137-400f-9252-e8574d94e34e
  Server: Kestrel
  x-ms-correlation-request-id: 418385f3-cea0-47b2-9dc6-82150a117b85
  x-ms-routing-request-id: WESTUS2:20200826T051719Z:418385f3-cea0-47b2-9dc6-82150a117b85
  X-Content-Type-Options: nosniff
  Date: Wed, 26 Aug 2020 05:17:19 GMT
  Content-Length: 688
  Content-Type: application/json; charset=utf-8
  Expires: -1
}'
BEFORE RESPONSE DISPATCH Value: '0'

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
VERBOSE: INFORMATION: BEFORE RESPONSE DISPATCH ExtendedData Type: ''
BEFORE RESPONSE DISPATCH ExtendedData ''

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
DEBUG: BeforeResponseDispatch: 

VERBOSE: INFORMATION: FINALLY The contents are 'Finally' and ''

Confirm
Continue with this operation?
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): 
VERBOSE: INFORMATION: FINALLY Parameter: ''
FINALLY RequestMessage 'Method: GET, RequestUri: 'https://management.azure.com/subscriptions/422b6c61-95b0-4213-b3be-7282315df71d/resourceGroups/arc-server-test/providers/Microsoft.HybridCompute/machines/tyler-ThinkPad-X1-Carbon/extensions/foo?api-version=2019-12-12', Version: 1.1, Content: <null>, Headers:
{
  x-ms-unique-id: 27
  x-ms-unique-id: 28
  x-ms-unique-id: 29
  x-ms-client-request-id: c7c5b66b-b95f-4d23-900b-24a34d04b320
  x-ms-client-request-id: c7c5b66b-b95f-4d23-900b-24a34d04b320
  x-ms-client-request-id: c7c5b66b-b95f-4d23-900b-24a34d04b320
  CommandName: Set-AzConnectedMachineExtension
  CommandName: Set-AzConnectedMachineExtension
  CommandName: Set-AzConnectedMachineExtension
  FullCommandName: Set-AzConnectedMachineExtension_Update
  FullCommandName: Set-AzConnectedMachineExtension_Update
  FullCommandName: Set-AzConnectedMachineExtension_Update
  ParameterSetName: __AllParameterSets
  ParameterSetName: __AllParameterSets
  ParameterSetName: __AllParameterSets
  User-Agent: AzurePowershell/Az4.0.0-preview
  User-Agent: AzurePowershell/Az4.0.0-preview
  User-Agent: AzurePowershell/Az4.0.0-preview
  Authorization: Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsIng1dCI6ImppYk5ia0ZTU2JteFBZck45Q0ZxUms0SzRndyIsImtpZCI6ImppYk5ia0ZTU2JteFBZck45Q0ZxUms0SzRndyJ9.eyJhdWQiOiJodHRwczovL21hbmFnZW1lbnQuY29yZS53aW5kb3dzLm5ldC8iLCJpc3MiOiJodHRwczovL3N0cy53aW5kb3dzLm5ldC83MmY5ODhiZi04NmYxLTQxYWYtOTFhYi0yZDdjZDAxMWRiNDcvIiwiaWF0IjoxNTk4NDE4MTQ4LCJuYmYiOjE1OTg0MTgxNDgsImV4cCI6MTU5ODQyMjA0OCwiX2NsYWltX25hbWVzIjp7Imdyb3VwcyI6InNyYzEifSwiX2NsYWltX3NvdXJjZXMiOnsic3JjMSI6eyJlbmRwb2ludCI6Imh0dHBzOi8vZ3JhcGgud2luZG93cy5uZXQvNzJmOTg4YmYtODZmMS00MWFmLTkxYWItMmQ3Y2QwMTFkYjQ3L3VzZXJzL2MyNTZjN2IwLWU5NGQtNDljNC04NTg1LTBjN2YzMDIwYmQzNy9nZXRNZW1iZXJPYmplY3RzIn19LCJhY3IiOiIxIiwiYWlvIjoiQVZRQXEvOFFBQUFBcnNiRnYvMU5kdXVpcjJuNy9nYUkzYkw4SkVNY3E1YnVlOEtZMGE3bk5VK3YwOFpIT1YzTG5EZGxqWkxrZUZRbk52OEJ5NUlMZkpIY3ErN29vaEFLQmx1bUV4Z1NNdis2OXRPVTRWa3pOcFE9IiwiYW1yIjpbInB3ZCIsIm1mYSJdLCJhcHBpZCI6IjE5NTBhMjU4LTIyN2ItNGUzMS1hOWNmLTcxNzQ5NTk0NWZjMiIsImFwcGlkYWNyIjoiMCIsImZhbWlseV9uYW1lIjoiTGVvbmhhcmR0IChQT1dFUlNIRUxMKSIsImdpdmVuX25hbWUiOiJUeWxlciIsImlwYWRkciI6Ijc1LjE3Mi4xODMuMTc2IiwibmFtZSI6IlR5bGVyIExlb25oYXJkdCAoUE9XRVJTSEVMTCkiLCJvaWQiOiJjMjU2YzdiMC1lOTRkLTQ5YzQtODU4NS0wYzdmMzAyMGJkMzciLCJvbnByZW1fc2lkIjoiUy0xLTUtMjEtMjEyNzUyMTE4NC0xNjA0MDEyOTIwLTE4ODc5Mjc1MjctMjIyNjkzMzgiLCJwdWlkIjoiMTAwMzdGRkU5OThDRkZGNyIsInJoIjoiMC5BUm9BdjRqNWN2R0dyMEdScXkxODBCSGJSMWlpVUJsN0lqRk9xYzl4ZEpXVVg4SWFBTk0uIiwic2NwIjoidXNlcl9pbXBlcnNvbmF0aW9uIiwic3ViIjoiMXJrT05YeDAtUDVJQnlFc2kyV1RROVZXQUtmNW9IQjRVRnNkV040aFlBSSIsInRpZCI6IjcyZjk4OGJmLTg2ZjEtNDFhZi05MWFiLTJkN2NkMDExZGI0NyIsInVuaXF1ZV9uYW1lIjoidHlsZW9uaGFAbWljcm9zb2Z0LmNvbSIsInVwbiI6InR5bGVvbmhhQG1pY3Jvc29mdC5jb20iLCJ1dGkiOiJZYzF3TWtsUUYwMkRiU1k5RXR3TEFBIiwidmVyIjoiMS4wIiwieG1zX3RjZHQiOjEyODkyNDE1NDd9.VbmeQcgINWvA1hIQ2IOZh1ND1QpY0qwo5UQm21Y2zXTJ6ah1PEJN4iSWOxk8KWsc7lpNB3BBEWtgoT4YDhE79kmFZl-miiw2CRTstzrYLSUVD4vG_yAhjJ6b7q3bgv_fnvioZ6TekJ7J9DNQWTLC_IVaD57-vnN8JDbG-GAD356Zl_ywBKGhZoJkoJPK-O_zb73SpP6kUw8uxa49UGqI3qt7VVAVYRIkSnPucLZu5VjwEBSGxzVpuUANh7Ag7Pz8k1ZXEupjTcsF2kcba3vVumtHm90PmpXoJlkJDQV8os7MpDaRJOcbc1p5FlzpiReL70HXxumZB2H5HmbCbucX1A
}'
FINALLY Response: 'StatusCode: 200, ReasonPhrase: 'OK', Version: 1.1, Content: System.Net.Http.HttpConnectionResponseContent, Headers:
{
  Cache-Control: no-cache
  Pragma: no-cache
  x-ms-ratelimit-remaining-subscription-reads: 11998
  Strict-Transport-Security: max-age=31536000; includeSubDomains
  x-ms-request-id: f68bd6e3-a137-400f-9252-e8574d94e34e
  Server: Kestrel
  x-ms-correlation-request-id: 418385f3-cea0-47b2-9dc6-82150a117b85
  x-ms-routing-request-id: WESTUS2:20200826T051719Z:418385f3-cea0-47b2-9dc6-82150a117b85
  X-Content-Type-Options: nosniff
  Date: Wed, 26 Aug 2020 05:17:19 GMT
  Content-Length: 688
  Content-Type: application/json; charset=utf-8
  Expires: -1
}'
```

Reply:
@TylerLeonhardt Please see the http request

```
HTTP Method:
PUT

Absolute Uri:
https://management.azure.com/subscriptions/422b6c61-95b0-4213-b3be-7282315df71d/resourceGroups/arc-server-test/providers/Microsoft.HybridCompute/machines/tyler-ThinkPad-X1-Carbon/extensions/foo?api-version=2019-12-12

Headers:

Body:
{
  "location": "eastus",
  "properties": {
    "instanceView": {
      "status": {
        "code": "success",
        "level": "Information",
        "message": "formattedMessage: Enable succeeded: \n[stdout]\nstderr\nstdout\n\n[stderr]\n"
      },
      "name": "foo",
      "type": "CustomScript",
      "typeHandlerVersion": "2.1.3"
    },
    "type": "CustomScript",
    "autoUpgradeMinorVersion": false,
    "publisher": "Microsoft.Azure.Extensions",
    "settings": {
      "commandToExecute": "ls -a"
    },
    "typeHandlerVersion": "2.1.3"
  }
}
```
Could you check in sever side whether the request can update commandToExecute correctly?

Reply:
Yep that's a good point. This should be working but it's not. I verified with the other parameter set:

```
PS /Users/tyleonha/Code/PowerShell/azure-powershell/src/ConnectedMachine [Az.ConnectedMachine]> $Settings = @{"commandToExecute" = "powershell.exe -c dir"}                                                                                   
PS /Users/tyleonha/Code/PowerShell/azure-powershell/src/ConnectedMachine [Az.ConnectedMachine]> Set-AzConnectedMachineExtension -Name food2 -ResourceGroupName csharp-sdk-test -MachineName TYLEONHA-DESK1 -Location eastus -Settings $settings

Name  Location ProvisioningState
----  -------- -----------------
food2 eastus   Succeeded

PS /Users/tyleonha/Code/PowerShell/azure-powershell/src/ConnectedMachine [Az.ConnectedMachine]> $c = Get-AzConnectedMachineExtension -ResourceGroupName csharp-sdk-test -MachineName TYLEONHA-DESK1                                           
PS /Users/tyleonha/Code/PowerShell/azure-powershell/src/ConnectedMachine [Az.ConnectedMachine]> $c.Setting

Name                           Value
----                           -----
commandToExecute               dir
```

 @edyoung @raghushantha can you assist here? Updating an extension's settings doesn't seem to be working.

Reply:
@msJinLei I also tried this scenario:

```pwsh
$ext = Get-AzConnectedMachineExtension -ResourceGroupName csharp-sdk-test -MachineName TYLEONHA-DESK1
$ext | Remove-AzConnectedMachineExtension
$ext |  Set-AzConnectedMachineExtension -MachineName TYLEONHA-DESK1 -ResourceGroupName csharp-sdk-test -Name foo
```

This yielded a resource not found:

```
    ResponseBody    : {"code":"HCRP404","message":"ResourceNotFound","target":"e6fe6705-4c9c-4b54-81d2-e455780e20b8/csharp-sdk-test/TYLEONHA-DESK1/foo"}
    Message         : ResourceNotFound
```

I still should try with 2 different machines but I fear the same result.

Reply:
I just tried this with two machines and it yielded the same result.

Reply:
For completeness, this also happens with `New-` and `Update-`.

Reply:
But this worked:

```
PS [Az.ConnectedMachine]> $Settings = @{"commandToExecute" = "powershell.exe -c dir"}                                                                                   
PS [Az.ConnectedMachine]> Update-AzConnectedMachineExtension -Name food2 -ResourceGroupName csharp-sdk-test -MachineName TYLEONHA-DESK1 -Settings $settings

Name  Location ProvisioningState
----  -------- -----------------
food2 eastus   Succeeded

PS [Az.ConnectedMachine]> $c = Get-AzConnectedMachineExtension -ResourceGroupName csharp-sdk-test -MachineName TYLEONHA-DESK1                                           
PS [Az.ConnectedMachine]> $c.Setting

Name                           Value
----                           -----
commandToExecute               powershell.exe -c dir
```

probably because this is doing a PATCH instead of a PUT?

Reply:
@msJinLei we've root caused this... it's because in the body of the request, the `typeHandlerVersion` is specified. How can I exclude this property from the body?

Reply:
maybe @isra-fel knows?

Reply:
Ok I have figured it out via implementing the partial `AfterToJson` method. That is pushed.